---
title: "Blood_healthy -Blood scRNAseq healthy state Oelen 2019"
output: html_notebook
---

The data used is collected from 120 Northern Netherlands population cohort Lifeline found from
https://eqtlgen.org/sc/datasets/1m-scbloodnl-dataset.html

For each Individual, the following data is collected:
Unstimulated
3h,24h stimulated with C. albicans
3h,24h stimulated with M. tuberculosis
3h,24h stimulated with P. aeruginosa

Two versions of the data are available:
v2: 907 genes/cell,
v3: 1861 genes/cell
The data is available in the following files:
1M_assignments_conditions_expid.tsv
1M_cell_types.tsv
1M_v2_20201029_RNA.mtx
1M_v2_20201029_RNA_colnames.tsv
1M_v2_20201029_RNA_rownames.tsv

Loading in the required packages and data
```{r load_packages, message=FALSE, warning=FALSE}
#Load the required packages
library(Seurat)
library(dplyr)
library(ggplot2)
library(patchwork)
library(rtracklayer)
library(methods)
library(Matrix)
library(monocle3)

# Load the data
# Set file paths
matrix_path <- "1M_v2_20201029_RNA.mtx"
cells_path <- "1M_v2_20201029_RNA_colnames.txt"
features_path <- "1M_v2_20201029_RNA_rownames.txt"
cell_types_path <- "1M_cell_types.tsv"
assignment_conditions <- "1M_assignments_conditions_expid.tsv"
#GTF file
gtf_file <- "Homo_sapiens.GRCh38.109.gtf"
# Read in counts matrix
counts <- readMM(matrix_path)
# Read in cells file and set column names of counts matrix
cells <- read.table(cells_path, header = FALSE)
colnames(counts) <- cells$V1
# Read in features file and set row names of counts matrix
features <- read.table(features_path, header = FALSE)
rownames(counts) <- features[,1]
#Read the annotation file
cell_types <- read.table(cell_types_path, header = TRUE, sep = "\t")
treatment <- read.table(assignment_conditions,header =TRUE ,sep ="\t")

```

Creating a Seurat object using the data to start analysing them.
```{r}
blood_healthy <- CreateSeuratObject(counts = counts, project = "blood_healthy", min.cells = 3, min.features = 200,Idents =cell_types$cell_type)
#saveRDS(blood_healthy,file="blood_healthy.rds")
blood_healthy<- readRDS("Reusable data/blood_healthy.rds")
```

Sorting the barcodes file to make sure they are in order and mapping them with the seurat object
```{r}
cell_types_ordered <- cell_types[match(colnames(blood_healthy), cell_types$barcode), ]
#Setting the Identity of the cells based on the barcode mapping
Idents(blood_healthy) <- cell_types_ordered$cell_type
#Uploading the meta data to the seurat object
blood_healthy[["barcode"]] <- cell_types_ordered$barcode
#Add cell type to meta data
blood_healthy[["cell_type"]] <- cell_types_ordered$cell_type
#Add treatment to meta data
treatment_ordered <- treatment[match(colnames(blood_healthy), treatment$barcode), ]
blood_healthy[["treatment"]] <- treatment_ordered$timepoint
blood_healthy[["subject"]] <- treatment_ordered$assignment
blood_healthy[["percent.mt"]] <- PercentageFeatureSet(blood_healthy, pattern = "^MT-")
```
The data is now ready to be analysed.We will start with the QC analysis after which we Normalise the data and find the variable features in the cells.
```{r}
#Subset the data with the following QC conditions
blood_healthy <- subset(blood_healthy, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 5)
#Normalize the data
blood_healthy <- NormalizeData(blood_healthy, normalization.method = "LogNormalize", scale.factor = 10000)
#Find variable features
blood_healthy <- FindVariableFeatures(blood_healthy, selection.method = "vst", nfeatures = 2000)
```
We will now visualise the variable features in the cells using the and Variable Feature Plot function.
```{r}
#Visualise the variable features in the cells using VariableFeaturePlot
top10<-head(VariableFeatures(blood_healthy),10)
plot1 <- VariableFeaturePlot(blood_healthy)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot2
#ggsave("VariableFeaturePlot.png",plot2)
```




We will now perform the PCA analysis to see the clustering of cells based on cell_types.
```{r}
#Scale the data
all.genes <- rownames(blood_healthy)
blood_healthy <- ScaleData(blood_healthy, features = all.genes)
#Perform PCA
blood_healthy <- RunPCA(blood_healthy, features = VariableFeatures(object = blood_healthy))
```
We will now visualise the PCA analysis using the PCAPlot function.
```{r}
#Visualise the PCA analysis using PCAPlot
pcaplot <- PCAPlot(blood_healthy, label = TRUE)
pcaplot
#ggsave("PCAPlot_bycell_type.png",pcaplot)
#Visualise the PCA analysis using VizDimLoadings
featuresdefpca <- VizDimLoadings(blood_healthy, dims = 1:4, reduction = "pca")
featuresdefpca
#ggsave("Defining_features.png",featuresdefpca)
Combined_features <- VizDimLoadings(blood_healthy, dims = 1:20, reduction = "pca",combine=TRUE)
Combined_features
#ggsave("Combined_features_plot.png",Combined_features)
```
Clustering the cells using umap
```{r}
plan(strategy = "multisession", workers = 50)
blood_healthy <- FindNeighbors(blood_healthy, dims = 1:20)
blood_healthy <- FindClusters(blood_healthy, resolution = 0.5,future.seed = TRUE)
blood_healthy <- RunUMAP(blood_healthy, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")
#Visualise the clustering using DimPlot
#DimPlot(blood_healthy, label = TRUE, group.by = "cell_type")
#ggsave("DimPlot_cell_type.png")
#DimPlot(blood_healthy, label = TRUE, group.by = "treatment")
#ggsave("DimPlot_treatment.png")
#DimPlot(blood_healthy, label = TRUE, group.by = c("treatment", "cell_type"))
#ggsave("DimPlot_treatment&cell_type.png")
```
We will now select the pDC cells and perform the analysis on them.
```{r}
#Select the pDC cells

# Subset the original object
pDC <- subset(blood_healthy, subset = cell_type == "pDC")

saveRDS(pDC,file="pDC_mod.rds")
pDC <-readRDS("pDC_mod.rds")
#Visualise the pDC cells using DimPlot
DimPlot(pDC, label = TRUE, group.by = "treatment")
ggsave("DimPlot_pDC_treatment.png")
DimPlot(pDC, label = TRUE)
ggsave("DimPlot_pDC_clustered.png")
```
We will now select the X and Y specific genes in the pDC cells to analyse the sex based differential expression.
```{r}
#Select the X and Y specific genes in the pDC cells
gtf <- rtracklayer::import(gtf_file)
gtf_df <- as.data.frame(gtf)

gene_rows <- gtf_df[gtf_df$seqnames %in% c("X", "Y") & gtf_df$type == "gene", ]

gene_info <- data.frame(gene_name = gene_rows$gene_name, chromosome = gene_rows$seqnames)

x_genes <- gene_info[gene_info$chromosome=="X",]
y_genes=gene_info[gene_info$chromosome=="Y",]

#Use x_genes and y_genes to filter the pDC object
gene_names <- rownames(pDC)
x_genes_present <- intersect(x_genes$gene_name, gene_names)
y_genes_present <- intersect(y_genes$gene_name, gene_names)

subset = gene_names %in% c(x_genes_present, y_genes_present)
matching_genes <- gene_names[subset]
#Selecting only the x amd y genes

pDC_meta <- subset(pDC, features = y_genes_present)
pDC_meta <- subset(pDC_meta, subset = nFeature_RNA > 0)
Idents(pDC_meta) <- "subject"


#Find genes in y_genes present whose Mean expression is >0
y_genes_expr <- AverageExpression(pDC_meta, features = y_genes_present,idents = "subject")
y_genes_mean_expr <- rowMeans(y_genes_expr$RNA)
#Get a vector of genes names with mean expression >0.5
y_genes_mean_expr <- y_genes_mean_expr > 0.5
#Select only the genes names with True value
y_genes_present_fil <- y_genes_present[y_genes_mean_expr]
#%%
# Calculate 95th percentile value for all the genes in y_genes_present
qval1 <- quantile(pDC_meta@assays$RNA@data["USP9Y",], probs = 0.9625)
qval2 <- quantile(pDC_meta@assays$RNA@data["EIF1AY",], probs = 0.9625)
qval5 <- quantile(pDC_meta@assays$RNA@data["KDM5D",], probs = 0.9625)
qval6 <- quantile(pDC_meta@assays$RNA@data["RPS4Y1",], probs = 0.9625)
qval7 <- quantile(pDC_meta@assays$RNA@data["ZFY",], probs = 0.9625)
qval9 <- quantile(pDC_meta@assays$RNA@data["DDX3Y",], probs = 0.9625)
qval10 <- quantile(pDC_meta@assays$RNA@data["UTY",], probs = 0.9625)
Male <- do.call("rbind", Male_list)

Male <- subset(pDC_meta, subset = USP9Y > qval1 | EIF1AY > qval2 | KDM5D > qval5 | RPS4Y1 > qval6 | ZFY > qval7 | DDX3Y > qval9 | UTY > qval10)
#%%
#Trying out the filter
qvals <- vector(length = length(y_genes_present_fil))
for (i in seq_along(y_genes_present_fil)) {
  qvals[i] <- quantile(pDC_meta@assays$RNA@data[y_genes_present_fil[i], ], probs = 0.95)
}

Male <-subset(pDC_meta,subset = EIF1AY > qvals[1] | RPS4Y1 > qvals[2])

#Add sex to the meta data
Male <- AddMetaData(Male, metadata = data.frame(sex = rep("male", nrow(Male@meta.data))))
Male@meta.data$sex <- "Male"
pDC[["sex"]]<-NA

#Now mark the subjects as Males in the pDC dataset
#Logic used If the subject values match in Male@meta.data$subject and pDC@meta.data$subject then add male to the corresponding sex column
# Find matching subjects in Male dataset
matching_subjects <- intersect(Male@meta.data$subject, pDC@meta.data$subject)

# Loop through the matching subjects and mark them as males in pDC dataset
for (subject in matching_subjects) {
  pDC@meta.data[pDC@meta.data$subject == subject, "sex"] <- "Male"
}
#Assigning the NA columns as female
pDC@meta.data$sex <- ifelse(is.na(pDC@meta.data$sex), "Female", pDC@meta.data$sex)
saveRDS(pDC,file="pDC_sex.rds")
pDC <- readRDS(file="pDC_sex.rds")
DimPlot(pDC)
#Analyse the pDC clusters
cells_per_subject <- table(pDC@meta.data$subject)

# Create a bar plot
barplot(cells_per_subject, main = "Cells per Person", xlab = "Person", ylab = "Number of Cells")
ggsave("Cells_per_person_bar.png")
#Visualize the clusters using bar plot group by sex
cells_per_sex <- table(pDC@meta.data$sex)
cell_counts_df <- data.frame(
  sex = names(cells_per_sex),
  count = as.numeric(cells_per_sex)
)

# Create the bar plot
ggplot(cell_counts_df, aes(x = sex, y = count, fill = sex)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("red", "blue")) +
  xlab("Sex") +
  ylab("Cell Count") +
  ggtitle("Number of Cells by Sex")
ggsave("cells_by_Sex_bar.png")

# Aggregate based on RNA_Cluster
pDC_sub <- subset(pDC, idents = c("17", "18", "19"))
# Create a table of cell counts by subject and cluster
cell_counts <- table(pDC_sub@meta.data$subject, pDC_sub@meta.data$seurat_clusters)

# Convert to a data frame and add a column for subject IDs
cell_counts_df <- as.data.frame.matrix(cell_counts)
cell_counts_df$subject <- rownames(cell_counts_df)

# Reshape the data to long format for plotting
cell_counts_long <- tidyr::gather(cell_counts_df, cluster, count, -subject)

# Create the stacked bar plot
ggplot(cell_counts_long, aes(x = subject, y = count, fill = cluster)) +
  geom_bar(stat = "identity") +
  xlab("Subject ID") +
  ylab("Cell Count") +
  ggtitle("Number of Cells by Cluster and Subject")
ggsave("cells_by_cluster_subject_bar.png")




```

```{r}
#Now split pDC by sexes
Male <- subset(pDC,subset= sex == "Male")
Female <-subset(pDC, subset =sex =="Female")

Male <-NormalizeData(Male)
Female <-NormalizeData(Female)

#Perform PCA on both the sexes
Male <- RunPCA(Male, features=VariableFeatures(object = Male))
Female <- RunPCA(Female,features =VariableFeatures(object = Female))
DimPlot(Male,reduction = "pca", group.by = "treatment")
ggsave("Male_pca_treat.png")
DimPlot(Female,reduction = "pca", group.by = "treatment")
ggsave("Female_pca_treat.png")
#Perform UMAP
plan(strategy = "multisession", workers = 50)
Male <- FindNeighbors(Male,dims=1:20)
Male <- FindClusters(Male, resolution = 0.5,future.seed = TRUE)
#Run UMAP
Male <- RunUMAP(Male, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")
#Visualize UMAP
DimPlot(Male,group.by = "treatment",label = TRUE)
ggsave("Male_treat_UMAP.png")
#Now visualize for Females
Female <-FindNeighbors(Female,dims=1:20)
Female <-FindClusters(Female,resolution = 0.5)
#Run UMAP
Female <- RunUMAP(Female, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")
DimPlot(Female,group.by = "treatment",label =TRUE)
ggsave("Female_treat_UMAP.png")

#Now performing PCA on the combined dataset
pDC <- NormalizeData(pDC)
pDC <- RunPCA(pDC, features = VariableFeatures(object = pDC))
DimPlot(pDC, reduction = "pca", group.by = "treatment",label = TRUE)
ggsave("pDC_treat_pca.png")
DimPlot(pDC,reduction = "pca",group.by = c("sex","treatment"),label = TRUE)
ggsave("pDC_sex&treat_pca.png")
#Perform UMAP
pDC <- FindNeighbors(pDC, dims = 1:20)
pDC <- FindClusters(pDC, resolution = 0.5)
pDC <- RunUMAP(pDC, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")
DimPlot(pDC,reduction = "umap",group.by= c("sex","treatment"),label = TRUE)
ggsave("pDC_sex&treat_UMAP.png")
DimPlot(pDC)
ggsave("pDC_UMAP_ungrouped.png")

Idents(pDC) <- "subject"
cluster.average <- AverageExpression(pDC,group.by="subject",return.seurat = TRUE)
# extract the numbers from the current idents
new_idents <- as.character(gsub("[^0-9]", "", names(Idents(cluster.average))))
# Add the new idents to the cluster.average object meta data
cluster.average[["subject"]] <- new_idents

# Create a lookup table of subject and sex and treatment values from the pDC dataset
subject_sex_lookup <- pDC@meta.data[, c("subject", "sex", "treatment")]
# Remove rows with NAs
subject_sex_lookup <- na.omit(subject_sex_lookup)
# Remove duplicates
subject_sex_lookup <- unique(subject_sex_lookup)
# Order by subject
subject_sex_lookup <- subject_sex_lookup[order(subject_sex_lookup$subject),]

# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average@meta.data$subject, subject_sex_lookup$subject)

# Fill in the "sex" column in the cluster.average object with the corresponding values from the pDC dataset
# Find the indices of matching rows in the cluster.average object
matching_rows <- match(matching_subjects, cluster.average@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
treatment_values <-subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "treatment"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average@meta.data$sex <- rep(NA, nrow(cluster.average@meta.data))
cluster.average@meta.data$treatment <- rep(NA, nrow(cluster.average@meta.data))
# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]
#Assign the treatment values to matching rows in the "treatment" column of cluster.average
cluster.average@meta.data$treatment[!is.na(matching_rows)] <- treatment_values[!is.na(matching_rows)]


#Normalize the cluster average
cluster.average <- NormalizeData(cluster.average)
#Scale the data
cluster.average <- ScaleData(cluster.average,features =rownames(cluster.average))
#Find variable features
cluster.average <- FindVariableFeatures(object=cluster.average)
#Run PCA
cluster.average <- RunPCA(cluster.average,features = VariableFeatures(object = cluster.average))
#Run UMAP
cluster.average <- FindNeighbors(cluster.average,dims=1:20)
#Find clusters
cluster.average <- FindClusters(cluster.average, resolution = 0.5)
#Run UMAP
cluster.average <- RunUMAP(cluster.average, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")



#Check for IFNA1

#DoHeatmap
DoHeatmap(cluster.average,features= c("EIF1AY","RPS4Y1"),group.by = 'sex')
ggsave("pDC_average&sex_heatmap.png")



```

```{r}
#Now perform DE analysis
#First perform DE analysis on the combined dataset
plan(strategy = "multisession", workers = 50)
Idents(cluster.average) <- cluster.average@meta.data$sex
markers <- FindAllMarkers(cluster.average, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
#Visualize the DE genes
VlnPlot(cluster.average,features = markers$gene)
ggsave("cluster.average_DE_genes.png")
VlnPlot(pDC,features = markers$gene,group.by = "sex")
ggsave("pDC_DE_genes_Vln.png")
DoHeatmap(pDC,features = markers$gene,group.by = "sex")
ggsave("pDC_DE_heatmap.png")
DoHeatmap(cluster.average,features = markers$gene, group.by ="sex")
ggsave("cluster.average_DE_heatmap.png")

Idents(pDC) <- pDC@meta.data$treatment
markers_treat <- FindAllMarkers(pDC, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
top10 <- head(markers_treat$gene,10)
VlnPlot(pDC,features = top10,group.by = "treatment")
ggsave("pDC_treat_DE_genes_Vln.png")
#VlnPlot for Interferon genes
#According to https://pubmed.ncbi.nlm.nih.gov/25839220/ IFNAR1, IFNAR2, LY6E, MX1, OAS3, IFI44L, IFI6, IFITM3 are upregulated which are all INF related genes.
VlnPlot(pDC,features = c("IFNAR1","IFNAR2","LY6E","MX1","OAS3","IFI44L","IFI6","IFITM3"),group.by = "treatment")
ggsave("pDC_treat_IFN_genes_Upreg_Vln.png")
#And these are the downregulated genes
#MGAM,ZNF331,PTGS2,MME,IL1R2,IL8,CXCL1,GOS2,NR4A2,NR4A3
VlnPlot(pDC,features = c("MGAM","ZNF331","PTGS2","MME","IL1R2","IL8","CXCL1","GOS2","NR4A2","NR4A3"),group.by = "treatment")
ggsave("pDC_treat_IFN_genes_Downreg_Vln.png")
```
```{r}
#Perform Pseudotime analysis using Monocle3 on the pDC Seurat object
#First convert the Seurat object to a Monocle3 object
 library(SeuratWrappers)
Idents(pDC) <- pDC@meta.data$treatment
pDC_mc <- as.cell_data_set(pDC)
recreate.partitions <- c(rep(1, length(pDC_mc@colData@rownames)))
names(recreate.partitions) <- pDC_mc@colData@rownames
recreate.partitions <- as.factor(recreate.partitions)
pDC_mc@clusters@listData[["UMAP"]][["partitions"]] <- recreate.partitions
list.cluster <- pDC@active.ident
pDC_mc@clusters@listData[["UMAP"]][["clusters"]] <- list.cluster
pDC_mc@int_colData@listData[["reducedDims"]]@listData[["UMAP"]] <- pDC@reductions$umap@cell.embeddings
cluster.before.traj <-plot_cells(pDC_mc, color_cells_by = "cluster", label_groups_by_cluster = F,
           group_label_size = 5) + theme(legend.position = "right")
cluster.before.traj
pDC_mc <- learn_graph(pDC_mc, use_partition = F)
plot_cells(pDC_mc, color_cells_by = "cluster", label_groups_by_cluster = F,
           group_label_size = 5) + theme(legend.position = "right")
ggsave("pDC_pseudotime_before.png")
pDC_mc <- order_cells(pDC_mc,reduction_method = "UMAP",verbose = T)
plot_cells(pDC_mc, color_cells_by = "pseudotime", label_groups_by_cluster = T,
           label_branch_points = T, label_roots = F, label_leaves = F) + theme(legend.position = "right")
ggsave("pDC._pseudotime.png")
```
```{r}
#Perform Gene set enrichment analysis and UMAP visualization using combined column
#Combine sex and subject column into a single column and add it to the meta data
library(EnhancedVolcano)
library(enrichplot)
library(clusterProfiler)
library(org.Hs.eg.db)
library(tibble)
library(plotly)

pDC_combined <- pDC
sexsubject <- paste(pDC_combined@meta.data$sex,pDC_combined@meta.data$subject,sep = "_")
sextreatment <- paste(pDC_combined@meta.data$sex,pDC_combined@meta.data$treatment,sep = "_")
pDC_combined[["sex.treatment"]] <- sextreatment
DimPlot(pDC_combined,group.by = "sex.treatment",label = T,pt.size = 0.5)
ggsave("Dimplot using combined column of sex andtreatment.png")

Idents(pDC_combined) <- pDC_combined@meta.data$sex.treatment
#Perform differential expression analysis on this combined column for each time point for each disease
unt_diffs <- FindMarkers(pDC_combined, ident.1 = "Male_UT", ident.2 = "Female_UT", min.pct = 0.25, logfc.threshold = 0)
#Volcano plot for Untreated Male vs Female
EnhancedVolcano(unt_diffs,x='avg_log2FC',y='p_val',lab=rownames(unt_diffs),pCutoff = 0.05,FCcutoff = 0.25)
ggsave("Volcano Plot Male_Female_UT.png")
#Gene Ontology Untreated
ego <- enrichGO(gene = rownames(unt_diffs) , OrgDb = org.Hs.eg.db, keyType = "SYMBOL", ont = "all", pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, readable = TRUE)
#Plot GO enrichment analysis
dotplot(ego, showCategory = 20)
ggsave("GO_enrichment_UT.png")
#Convert to Entez IDs
Ids_entrez <- mapIds(org.Hs.eg.db,keys=rownames(unt_diffs),column="ENTREZID", keytype="SYMBOL", multiVals="first")
#Perform KEGG enrichment analysis
kegg <- enrichKEGG(gene = Ids_entrez, organism = "hsa", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
dotplot(kegg, showCategory = 20)
ggsave("KEGG_enrichment_UT.png")


#3H Candida
CA3h_male_diffs <- FindMarkers(pDC_combined, ident.1 = "Male_3hCA", ident.2 = "Male_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(CA3h_male_diffs) <- paste(colnames(CA3h_male_diffs),"Male",sep = "_")
CA3h_female_diffs <- FindMarkers(pDC_combined, ident.1 = "Female_3hCA", ident.2 = "Female_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(CA3h_female_diffs) <- paste(colnames(CA3h_female_diffs),"Female",sep = "_")
#Combine the two dataframes
merged_data <- merge(CA3h_male_diffs, CA3h_female_diffs, by="row.names", all=TRUE)
plot1_3h <- ggplot(merged_data, aes(x = avg_log2FC_Male, y = avg_log2FC_Female, label = Row.names)) +
  geom_point() +
  geom_text(hjust = -0.1, vjust = 0.1,size=3) +
  geom_abline(intercept = 0, slope = 1, color = "red") +
  theme_bw()
plot1_3h
ggsave(filename = "CA_3H.png",plot = plot1_3h)
ggplotly(plot1_3h)
#Filter marker based on p value and log2fc
CA3h_male_de_genes <- CA3h_male_diffs[CA3h_male_diffs$p_val_Male < 0.01 & abs(CA3h_male_diffs$avg_log2FC_Male) > 0.50, ]
CA3h_female_de_genes <- CA3h_female_diffs[CA3h_female_diffs$p_val_Female < 0.01 & abs(CA3h_female_diffs$avg_log2FC_Female) > 0.50, ]
genes_CA_3h <- c(rownames(CA3h_female_de_genes),rownames(CA3h_male_de_genes))

#Perform GO enrichment analysis
ego <- enrichGO(gene = genes_CA_3h , OrgDb = org.Hs.eg.db, keyType = "SYMBOL", ont = "all", pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, readable = TRUE)
#Plot GO enrichment analysis
dotplot(ego, showCategory = 20)
ggsave("GO_enrichment_CA_3H.png")
#Convert to Entez IDs
Ids_entrez <- mapIds(org.Hs.eg.db,keys=genes_CA_3h,column="ENTREZID", keytype="SYMBOL", multiVals="first")
#Perform KEGG enrichment analysis
kegg <- enrichKEGG(gene = Ids_entrez, organism = "hsa", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
dotplot(kegg, showCategory = 20)
ggsave("KEGG_enrichment_CA_3H.png")

#24H Candida

CA24h_male_diffs <- FindMarkers(pDC_combined, ident.1 = "Male_24hCA", ident.2 = "Male_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(CA24h_male_diffs) <- paste(colnames(CA24h_male_diffs),"Male",sep = "_")
CA24h_female_diffs <-FindMarkers(pDC_combined,ident.1 = "Female_24hCA",ident.2 = "Female_UT",min.pct = 0.25,logfc.threshold = 0.25)
colnames(CA24h_female_diffs) <- paste(colnames(CA24h_female_diffs),"Female",sep = "_")
merged_data <- merge(CA24h_male_diffs, CA24h_female_diffs, by="row.names", all=TRUE)
plot1_24h <-ggplot(merged_data, aes(x = avg_log2FC_Male, y = avg_log2FC_Female, label = Row.names)) +
  geom_point() +
  geom_text(hjust = -0.1, vjust = 0.1,size = 3) +
  geom_abline(intercept = 0, slope = 1, color = "red") +
  theme_bw()
plot1_24h
ggsave(filename = "CA_24H.png",plot = plot1_24h)
ggplotly(plot1_24h)
#Filter marker based on p value and log2fc
CA24h_male_de_genes <- CA24h_male_diffs[CA24h_male_diffs$p_val_Male < 0.01 & abs(CA24h_male_diffs$avg_log2FC_Male) > 0.50, ]
CA24h_female_de_genes <- CA24h_female_diffs[CA24h_female_diffs$p_val_Female < 0.01 & abs(CA24h_female_diffs$avg_log2FC_Female) > 0.50, ]
genes_CA_24h <- c(rownames(CA24h_female_de_genes),rownames(CA24h_male_de_genes))

#Perform GO enrichment analysis
ego <- enrichGO(gene = genes_CA_24h , OrgDb = org.Hs.eg.db, keyType = "SYMBOL", ont = "all", pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, readable = TRUE)
#Plot GO enrichment analysis
dotplot(ego, showCategory = 20)
ggsave(filename = "CA_24H_GO.png",plot = dotplot(ego, showCategory = 20))
#Convert to Entez IDs
Ids_entrez <- mapIds(org.Hs.eg.db,keys=genes_CA_24h,column="ENTREZID", keytype="SYMBOL", multiVals="first")
#Perform KEGG enrichment analysis
kegg <- enrichKEGG(gene = Ids_entrez, organism = "hsa", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
dotplot(kegg, showCategory = 20)
ggsave(filename = "CA_24H_KEGG.png",plot = dotplot(kegg, showCategory = 20))

#3h Pseudomonas
PA3h_male_diffs <- FindMarkers(pDC_combined, ident.1 = "Male_3hPA", ident.2 = "Male_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(PA3h_male_diffs) <- paste(colnames(PA3h_male_diffs),"Male",sep = "_")
PA3h_female_diffs <- FindMarkers(pDC_combined, ident.1 = "Female_3hPA", ident.2 = "Female_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(PA3h_female_diffs) <- paste(colnames(PA3h_female_diffs),"Female",sep = "_")
#Combine the two dataframes
merged_data <- merge(PA3h_male_diffs, PA3h_female_diffs, by="row.names", all=TRUE)
plot2_3h <- ggplot(merged_data, aes(x = avg_log2FC_Male, y = avg_log2FC_Female, label = Row.names)) +
  geom_point() +
  geom_text(hjust = -0.1, vjust = 0.1) +
  geom_abline(intercept = 0, slope = 1, color = "red") +
  theme_bw()
plot2_3h
ggsave(filename = "PA_3H.png",plot = plot2_3h)
ggplotly(plot2_3h)

#Filter marker based on p value and log2fc
PA3h_male_de_genes <- PA3h_male_diffs[PA3h_male_diffs$p_val_Male < 0.01 & abs(PA3h_male_diffs$avg_log2FC_Male) > 0.50, ]
PA3h_female_de_genes <- PA3h_female_diffs[PA3h_female_diffs$p_val_Female < 0.01 & abs(PA3h_female_diffs$avg_log2FC_Female) > 0.50, ]
genes_PA_3h <- c(rownames(PA3h_female_de_genes),rownames(PA3h_male_de_genes))

#Perform GO enrichment analysis
ego <- enrichGO(gene = genes_PA_3h , OrgDb = org.Hs.eg.db, keyType = "SYMBOL", ont = "all", pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, readable = TRUE)
#Plot GO enrichment analysis
dotplot(ego, showCategory = 20)
ggsave(filename = "PA_3H_GO.png",plot = dotplot(ego, showCategory = 20))
#Convert to Entez IDs
Ids_entrez <- mapIds(org.Hs.eg.db,keys=genes_PA_3h,column="ENTREZID", keytype="SYMBOL", multiVals="first")
#Perform KEGG enrichment analysis
kegg <- enrichKEGG(gene = Ids_entrez, organism = "hsa", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
dotplot(kegg, showCategory = 20)
ggsave(filename = "PA_3H_KEGG.png",plot = dotplot(kegg, showCategory = 20))
#24h Pseudomonas
PA24h_male_diffs <- FindMarkers(pDC_combined, ident.1 = "Male_24hPA", ident.2 = "Male_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(PA24h_male_diffs) <- paste(colnames(PA24h_male_diffs),"Male",sep = "_")
PA24h_female_diffs <-FindMarkers(pDC_combined,ident.1 = "Female_24hPA",ident.2 = "Female_UT",min.pct = 0.25,logfc.threshold = 0.25)
colnames(PA24h_female_diffs) <- paste(colnames(PA24h_female_diffs),"Female",sep = "_")
merged_data <- merge(PA24h_male_diffs, PA24h_female_diffs, by="row.names", all=TRUE)
plot2_24h <-ggplot(merged_data, aes(x = avg_log2FC_Male, y = avg_log2FC_Female, label = Row.names)) +
  geom_point() +
  geom_text(hjust = -0.1, vjust = 0.1) +
  geom_abline(intercept = 0, slope = 1, color = "red") +
  theme_bw()
plot2_24h
ggsave(filename = "PA_24H.png",plot = plot2_24h)
ggplotly(plot2_24h)

#Filter marker based on p value and log2fc
PA24h_male_de_genes <- PA24h_male_diffs[PA24h_male_diffs$p_val_Male < 0.01 & abs(PA24h_male_diffs$avg_log2FC_Male) > 0.50, ]
PA24h_female_de_genes <- PA24h_female_diffs[PA24h_female_diffs$p_val_Female < 0.01 & abs(PA3h_female_diffs$avg_log2FC_Female) > 0.50, ]
genes_PA_24h <- c(rownames(PA24h_female_de_genes),rownames(PA24h_male_de_genes))

#Perform GO enrichment analysis
ego <- enrichGO(gene = genes_PA_24h , OrgDb = org.Hs.eg.db, keyType = "SYMBOL", ont = "all", pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, readable = TRUE)
#Plot GO enrichment analysis
dotplot(ego, showCategory = 20)
ggsave(filename = "PA_24H_GO.png",plot = dotplot(ego, showCategory = 20))
#Convert to Entez IDs
Ids_entrez <- mapIds(org.Hs.eg.db,keys=genes_PA_24h,column="ENTREZID", keytype="SYMBOL", multiVals="first")
#Perform KEGG enrichment analysis
kegg <- enrichKEGG(gene = Ids_entrez, organism = "hsa", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
dotplot(kegg, showCategory = 20)
ggsave(filename = "PA_24H_KEGG.png",plot = dotplot(kegg, showCategory = 20))


#3H M.Tuberculosis
MTB3h_male_diffs <- FindMarkers(pDC_combined, ident.1 = "Male_3hMTB", ident.2 = "Male_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(MTB3h_male_diffs) <- paste(colnames(MTB3h_male_diffs),"Male",sep = "_")
MTB3h_female_diffs <- FindMarkers(pDC_combined, ident.1 = "Female_3hMTB", ident.2 = "Female_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(MTB3h_female_diffs) <- paste(colnames(MTB3h_female_diffs),"Female",sep = "_")
#Combine the two dataframes
merged_data <- merge(MTB3h_male_diffs, MTB3h_female_diffs, by="row.names", all=TRUE)
plot3_3h <- ggplot(merged_data, aes(x = avg_log2FC_Male, y = avg_log2FC_Female, label = Row.names)) +
  geom_point() +
  geom_text(hjust = -0.1, vjust = 0.1) +
  geom_abline(intercept = 0, slope = 1, color = "red") +
  theme_bw()
plot3_3h
ggsave(filename = "MTB_3H.png",plot = plot3_3h)
ggplotly(plot3_3h)

#Filter marker based on p value and log2fc
MTB3h_male_de_genes <- MTB3h_male_diffs[MTB3h_male_diffs$p_val_Male < 0.01 & abs(MTB3h_male_diffs$avg_log2FC_Male) > 0.50, ]
MTB3h_female_de_genes <- MTB3h_female_diffs[MTB3h_female_diffs$p_val_Female < 0.01 & abs(MTB3h_female_diffs$avg_log2FC_Female) > 0.50, ]
genes_MTB_3h <- c(rownames(MTB3h_female_de_genes),rownames(MTB3h_male_de_genes))

#Perform GO enrichment analysis
ego <- enrichGO(gene = genes_MTB_3h , OrgDb = org.Hs.eg.db, keyType = "SYMBOL", ont = "all", pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, readable = TRUE)
#Plot GO enrichment analysis
dotplot(ego, showCategory = 20)
ggsave(filename = "MTB_3H_GO.png",plot = dotplot(ego, showCategory = 20))
#Convert to Entez IDs
Ids_entrez <- mapIds(org.Hs.eg.db,keys=genes_MTB_3h,column="ENTREZID", keytype="SYMBOL", multiVals="first")
#Perform KEGG enrichment analysis
kegg <- enrichKEGG(gene = Ids_entrez, organism = "hsa", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
dotplot(kegg, showCategory = 20)
ggsave(filename = "MTB_3H_KEGG.png",plot = dotplot(kegg, showCategory = 20))
#24H M.Tuberculosis
MTB24h_male_diffs <- FindMarkers(pDC_combined, ident.1 = "Male_24hMTB", ident.2 = "Male_UT", min.pct = 0.25, logfc.threshold = 0.25)
colnames(MTB24h_male_diffs) <- paste(colnames(MTB24h_male_diffs),"Male",sep = "_")
MTB24h_female_diffs <-FindMarkers(pDC_combined,ident.1 = "Female_24hMTB",ident.2 = "Female_UT",min.pct = 0.25,logfc.threshold = 0.25)
colnames(MTB24h_female_diffs) <- paste(colnames(MTB24h_female_diffs),"Female",sep = "_")
merged_data <- merge(MTB24h_male_diffs, MTB24h_female_diffs, by="row.names", all=TRUE)
plot3_24h <-ggplot(merged_data, aes(x = avg_log2FC_Male, y = avg_log2FC_Female, label = Row.names)) +
  geom_point() +
  geom_text(hjust = -0.1, vjust = 0.1) +
  geom_abline(intercept = 0, slope = 1, color = "red") +
  theme_bw()
plot3_24h
ggsave(filename = "MTB_24H.png",plot = plot3_24h)
ggplotly(plot3_24h)

#Filter marker based on p value and log2fc
MTB24h_male_de_genes <- MTB24h_male_diffs[MTB24h_male_diffs$p_val_Male < 0.01 & abs(MTB24h_male_diffs$avg_log2FC_Male) > 0.50, ]
MTB24h_female_de_genes <- MTB24h_female_diffs[MTB24h_female_diffs$p_val_Female < 0.01 & abs(MTB24h_female_diffs$avg_log2FC_Female) > 0.50, ]
genes_MTB_24h <- c(rownames(MTB24h_female_de_genes),rownames(MTB24h_male_de_genes))

#Perform GO enrichment analysis
ego <- enrichGO(gene = genes_MTB_24h , OrgDb = org.Hs.eg.db, keyType = "SYMBOL", ont = "all", pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, readable = TRUE)
#Plot GO enrichment analysis
dotplot(ego, showCategory = 20)
ggsave(filename = "MTB_24H_GO.png",plot = dotplot(ego, showCategory = 20))
#Convert to Entez IDs
Ids_entrez <- mapIds(org.Hs.eg.db,keys=genes_MTB_24h,column="ENTREZID", keytype="SYMBOL", multiVals="first")
#Perform KEGG enrichment analysis
kegg <- enrichKEGG(gene = Ids_entrez, organism = "hsa", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
dotplot(kegg, showCategory = 20)
ggsave(filename = "MTB_24H_KEGG.png",plot = dotplot(kegg, showCategory = 20))
```


