---
title: "Visualizing GO and KEGG Expression through Heatmap"
output: html_document
---

```{r}
library(paletteer)
library(RColorBrewer)
#Subset the cluster.average to select only the UT individuals
UT <- subset(pDC,subset = treatment == 'UT')
cluster.average_UT <-AverageExpression(UT,group.by = 'subject',return.seurat = T)
#Normalize the data
cluster.average_UT <- NormalizeData(cluster.average_UT)
#Scale the data
cluster.average_UT <- ScaleData(cluster.average_UT)
#Find Variable Genes
cluster.average_UT <- FindVariableFeatures(cluster.average_UT)
#Perform a PCA
cluster.average_UT <- RunPCA(cluster.average_UT,dims=1:20)
#Find Neighbors
cluster.average_UT <- FindNeighbors(cluster.average_UT)
#Find Clusters
cluster.average_UT <- FindClusters(cluster.average_UT,resolution = 0.5)

#Run UMAP
cluster.average_UT <- RunUMAP(cluster.average_UT, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")

subject_sex_lookup <- pDC@meta.data[, c("subject", "sex")]
# Remove rows with NAs
subject_sex_lookup <- na.omit(subject_sex_lookup)
# Remove duplicates
subject_sex_lookup <- unique(subject_sex_lookup)
# Order by subject
subject_sex_lookup <- subject_sex_lookup[order(subject_sex_lookup$subject),]
#Add subject column to cluster.average_UT
cluster.average_UT@meta.data$subject <-names(Idents(cluster.average_UT))
# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average_UT@meta.data$subject, subject_sex_lookup$subject)

# Fill in the "sex" column in the cluster.average object with the corresponding values from the pDC dataset
# Find the indices of matching rows in the cluster.average object
matching_rows <- match(matching_subjects, cluster.average_UT@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average_UT@meta.data$sex <- rep(NA, nrow(cluster.average_UT@meta.data))

# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average_UT@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]

my_palette <- paletteer::paletteer_c("grDevices::Blue-Red", n = 20, direction = -1)

#GO-nitric oxide metabolic process(Gene list from the core_enrichment genes)
DoHeatmap(cluster.average_UT,features=c("TSPO","KLF4","HSP90AA1","CD36","IL1B","AIF1","SOD2"),group.by = 'sex',draw.lines = T,)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))


DoHeatmap(pDC_combined,features = c("TSPO","KLF4","HSP90AA1","CD36","IL1B","AIF1","SOD2"),group.by = 'sex.treatment',disp.min = -1, disp.max = 1)+ scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Using all the genes from the pathway
DoHeatmap(cluster.average_UT,features = c("TLR2", "NOS1", "CYP1B1", "CYB5R3", "DDAH2", "NQO1", "GCH1", "CYB5B", "NOS3", "ARG2", "MTARC1", "TLR4", "TICAM1", "SLC7A6", "KLRK1", "AKT1", "POR", "CPS1", "MTARC2", "RORA", "GCHFR", "ACP5", "SPR", "NOS2", "CAV1"),group.by = 'sex',disp.min = -1,disp.max = 1)

#KEGG Carbon metabolism
DoHeatmap(cluster.average_UT,features = c("GRSF1","OCIAD1","RAB21","SOCS1","PTGDS","CSF2RB","MARCH1"),group.by = 'sex')+scale_fill_gradientn(colors = paletteer::paletteer_c("grDevices::Blue-Red", n=20, direction = -1), oob=scales::oob_squish_any, limits=c(-2,2))


DoHeatmap(pDC_combined,features = c("GRSF1","OCIAD1","RAB21","SOCS1","PTGDS","CSF2RB","MARCH1"),group.by = 'sex')
#Glycolysis
DoHeatmap(cluster.average_UT,features = c("GAPDH","ADH5","ADPGK","PGK1","LDHA","ALDH3B1","MINPP1","PGAM1","PFKM","PFKL","ALDH9A1","ALDH2","PKM","GPI","ALDH7A1","PGAM4","FBP1","ALDOA","ENO2","LDHB","ALDH1B1","HK3"),group.by = 'sex',disp.min = -1,disp.max = 1)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#TCA cycle
DoHeatmap(cluster.average_UT,features = c("PDHB","MDH2","OGDH","SDHC","FH","DLAT","ACO2","SDHA","IDH3G","ACLY","CS","SUCLG1","DLD","SDHD","IDH3A"),group.by = 'sex',disp.min = -2,disp.max = 2)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

# Oxidative phosphorylation
DoHeatmap(cluster.average_UT,features = c("NDUFA11","COX5B","ATP5ME","COX8A","UQCRQ","NDUFB1","NDUFA1","ATP5MC2","NDUFS6","COX7C","UQCR11","COX6B1","NDUFB3","ATP5MG","UQCRH","COX7B2"),group.by = 'sex')+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-3, 3))

#Upregulation of Microphilial/Macrophage activation_GO
DoHeatmap(cluster.average_UT,features = c("CCL3", "STAP1","CTSC","CCL3","MIR142"),group.by = 'sex') + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#IL-6 Production
DoHeatmap(cluster.average_UT,features =c("ARRB2","TNFAIP3","PYCARD","TYROBP","CD36","IL1B","AIF1"),group.by = 'sex',disp.min = -1,disp.max = 1)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#IL-17 Production
DoHeatmap(cluster.average_UT,features=c("RFTN1","SLC7A5","IFNG" ,"LILRB2", "AIF1",  "AKIRIN2" ,"CD36", "EREG", "ZBTB20", "HYAL2" ,"APP" ,"TSLP" ,"F2R","PTAFR","SETD4","NOS2","TLR3","HMGB1"),group.by='sex',disp.min = -1,disp.max = 1)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#Carbon Metabolism-KEGG
DoHeatmap(cluster.average_UT,features = c("GRSF1","OCIAD1","RAB21","SOCS1","PTGDS","CSF2RB","MARCH1" ),group.by = 'sex',disp.min = -2,disp.max = 2)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Biosynthesis of amino acids-KEGG
DoHeatmap(cluster.average_UT,features = c("GRSF1","OCIAD1","RAB21","CSF2RB","MARCH1"),group.by = 'sex',disp.min = -1,disp.max = 1)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#Cholesterol Metabolism-KEGG
DoHeatmap(cluster.average_UT,features = c("CCT3","HNRNPA2B1","LIMD2","RPL28"),group.by = 'sex',disp.min = -1.5,disp.max = 1.5)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#Pentose Phosphate Pathway-KEGG

DoHeatmap(cluster.average_UT, features = c("OCIAD1", "RAB21", "PTGDS"),
          group.by = 'sex', disp.min = -1, disp.max = 1) +
  scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
```

position the caret at any line or the code chunk, then click "+".

The code chunk appears:
```{r}
#CA
# Subset pDC_CA
pDC_CA <-subset(pDC_combined,subset= treatment == '3hCA')
cluster.average_3hCA <-AverageExpression(pDC_CA,group.by = 'subject' ,return.seurat= T)
#Normalize the data
cluster.average_3hCA <- NormalizeData(cluster.average_3hCA)
#Scale the data
cluster.average_3hCA <- ScaleData(cluster.average_3hCA)
#Find variable genes
cluster.average_3hCA <- FindVariableFeatures(cluster.average_3hCA)
#Run PCA
cluster.average_3hCA <- RunPCA(cluster.average_3hCA,dims=1:20)
#Find Neighbors
cluster.average_3hCA <- FindNeighbors(cluster.average_3hCA)
#Find Clusters
cluster.average_3hCA <- FindClusters(cluster.average_3hCA, resolution = 0.5)
#Run UMAP
cluster.average_3hCA <- RunUMAP(cluster.average_3hCA, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")
#Add the sex and subject information.
subject_sex_lookup <- pDC@meta.data[, c("subject", "sex")]
# Remove rows with NAs
subject_sex_lookup <- na.omit(subject_sex_lookup)
# Remove duplicates
subject_sex_lookup <- unique(subject_sex_lookup)
# Order by subject
subject_sex_lookup <- subject_sex_lookup[order(subject_sex_lookup$subject),]
#Add subject column to cluster.average_3hCA
cluster.average_3hCA@meta.data$subject <-names(Idents(cluster.average_3hCA))
# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average_3hCA@meta.data$subject, subject_sex_lookup$subject)

# Fill in the "sex" column in the cluster.average object with the corresponding values from the pDC dataset
# Find the indices of matching rows in the cluster.average object
matching_rows <- match(matching_subjects, cluster.average_3hCA@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average_3hCA@meta.data$sex <- rep(NA, nrow(cluster.average_3hCA@meta.data))

# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average_3hCA@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]

# Male
#Regulation of Viral genome replication-GO
DoHeatmap(cluster.average_3hCA,features = c("ISG15","ISG20","MX1","IFI16","PLSCR1","OAS2","EIF2AK2","IFIH1","ADAR","OAS1"),group.by = 'sex',disp.min = -1,disp.max = 1)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Using all the genes for the gene ontology
DoHeatmap(cluster.average_3hCA, features = c("DDB1", "PKN2", "PPIE", "AICDA", "TRIM6", "OAS2", "IFI16", "APOBEC3B", "PPIA", "ZNFX1", "HMGA2", "CNOT7", "CD28", "BANF1", "VAPB", "STAU1", "BCL2", "ISG20", "CXCL8", "TASOR", "MX1", "N4BP1", "IFIT5", "OAS1", "FAM111A", "TMEM39A", "EIF2AK2", "IFNB1", "HACD3", "LARP1", "TRIM28", "SRPK1", "GBP7", "RAD23A", "APOBEC3C", "APOBEC3F", "FKBP6", "PPARA", "CCL5", "RESF1", "SHFL", "TOP2A", "TRIM38", "MPHOSPH8", "SETDB1", "TNIP1", "IFIH1", "ZC3HAV1", "APOBEC3D", "ADARB1", "ADAR", "PABPC1", "APOBEC3G", "IFITM1", "NOTCH1", "IFIT1", "APOBEC3H", "SRPK2", "DDX3X", "RSAD2", "BST2", "RNASEL", "PDE12", "BTBD17", "MORC2", "IFNL3", "TNF", "OAS3", "PPIH", "MAVS", "APOBEC3A", "PROX1", "OASL", "TARBP2", "LTF", "ISG15", "SLPI", "TOP2B", "ILF3", "INPP5K", "PPID", "PLSCR1", "IFITM2", "IFITM3"), group.by = 'sex', disp.min = -2, disp.max = 2)

#Defense response to virus-GO
DoHeatmap(cluster.average_3hCA,features = c("CXCL10","ISG15","PMAIP1","BIRC3","ISG20","IFIT2","MX1","IFIT3","IFI44L","IFI16","PLSCR1","STAT1","OAS2","EIF2AK2","NT5C3A","IFIH1","ADAR","HERC5","IRF1","IFI6","TANK","OAS1","MX2"),group.by = 'sex',disp.min = -1,disp.max = 1)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#All the genes
gene_list <- c("IFNL4", "DDX60L", "TLR2", "CARD9", "CASP1", "IL12B", "DDX21", "DHX15", "FADD", "TRAF3", "DHX58", "FCN3", "TBK1", "TRIM6", "OAS2", "TRIM52", "DHX36", "IFI16", "LILRB1", "AGBL4", "TBKBP1", "APOBEC3B", "DMBT1", "DEFA3", "DEFA1", "GBP2", "GBP1", "NT5C3A", "ATG7", "RIGI", "RTP4", "ISG20", "HERC5", "UBE2W", "CD207", "RELA", "BECN1", "IL6", "MLKL", "AZU1", "PYCARD", "DDIT4", "RIPK3", "STAT1", "ZBP1", "CD40", "IRF3", "IFIT5", "TLR7", "TRIM27", "UNC13D", "SLFN13", "DHX16", "OAS1", "DDX60", "IL10RB", "ABCF3", "EIF2AK2", "TRIM56", "IFNA2", "IFNA10", "IFNA21", "IFNA5", "IFNA17", "IFNB1", "TRIM28", "GBP7", "UBE2N", "PRF1", "PML", "IKBKE", "APOBEC3C", "AZI2", "APOBEC3F", "GARIN5A", "UNC93B1", "TRIM8", "STAT2", "BCL2L1", "STING1", "URS0000338542_9606", "CLPB", "sting_human-1", "alt-cbc_human", "MYD88", "MAP3K14", "TRIM22", "IFNE", "USP44", "NLRC5", "IFNL1", "IFIH1", "SETD2", "EIF2AK4", "IRF9", "ZC3HAV1", "NLRP9", "USP29", "APOBEC3D", "TANK", "IRF5", "ADARB1", "ADAR", "RNF185", "APOBEC3G", "GBP3", "TRIM41", "IFNGR2", "MBL2", "RSAD2", "BST2", "GPAM", "TLR8", "IFNK", "IRF7", "IFNL2", "IFNL3", "TNF", "OAS3", "NLRP6", "IFNG", "CARD8", "F2RL1", "MAVS", "USP20", "PARP9", "APOBEC3A", "OASL", "IFNAR2", "TRIM26", "IFNA16", "IFNW1", "EXOSC5", "POLR3A", "GBP5", "IFI27", "IRF2")

DoHeatmap(cluster.average_3hCA, features = gene_list, group.by = 'sex', disp.min = -1, disp.max = 1)

#Negative Regulation of Immune System Process-GO
DoHeatmap(cluster.average_3hCA,features=c("IL2RA","IL7R","ISG15","AHR","IFI16","PARP14","IL4I1","ADTRP","LPXN","PAG1","CCL3","HLA-A","ADAR","IRF1","HLA-F","RUNX3","CD47","USP18","OAS1","HLA-E","CASP3","CCL2","HLA-DRB1","SERPINB9","SAMSN1","BTN2A2","NMI","TNFAIP3","ELF1"),group.by = 'sex',disp.min = -1,disp.max = 1)+scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#Cell Death-GO
DoHeatmap(cluster.average_3hCA, features = c("CXCL10","OPN3","CTSC","TXN","BCL2A1","IL2RA","ENDOG","IL7R","CFLAR","BTG1","GCLM","SLC7A11","NFKB1","CDKN1A","PMAIP1","BIRC3","IFIT2","MX1","IFIT3","SQSTM1","AHR","IFI16","PLSCR1","TNFSF10","PTPN1","BID","ATF5","STAT1","TNFAIP8","EIF2AK2","AKR1C3","KMO","MARCKS","CCR7","ARL6IP5","GZMB","CCL3","NINJ1","SRGN","XAF1","STK4","ADAR","EMP3","IRF1","IFI6","HLA-F","TAF9","CASP3","TRAF1","PKM","SP100","DNAJA1","CCL2","CD38","ACTN4","FAS","NUP62","ATF4","SERPINB9","RTN4","RBCK1","HSP90B1"), group.by = 'sex', disp.min = -1, disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Cell Migration-GO
DoHeatmap(cluster.average_3hCA, features = c("SMPD3", "CCL5", "AIF1", "B4GALT1", "CORO1A", "MAP2K3", "RAC2", "SELPLG", "PHACTR1", "SERPINF1", "GPX1", "LDLRAD4", "DUSP1", "PPBP", "PYCARD", "ACTG1", "KLF4", "ARHGDIB", "S100A8", "NR4A2", "NR4A1", "S100A9", "MYADM", "ID1", "RGCC", "CXCR4", "FCER1G"), group.by = "sex", disp.min = -1, disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Skeletal Muscle cell differentitaion-GO
DoHeatmap(cluster.average_3hCA,features = c("NR4A1","FOS"),group.by="sex",disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Using All the genes
DoHeatmap(cluster.average_3hCA,features =c("HIVEP3","MED20","SAP30","LEMD2","GTF3C5","NR4A1","SMYD1","MAFF","MEF2D","MEGF10","HEYL","PHOX2B","VAX1","SELENON","BCL9L","BTG2","ANKRD33","FOXN2","KRAS","CDON","FOS","DMRTA2","MYF6","KLF5","SCX","NOTCH1","MYF5","WNT3A","CITED2","SHH","ATF3","HMG20B","KLHL40","BCL9","ZNF689","EGR1","EGR2","RB1","EMD","PAX5","BARX2","SOX11","SOX8","MSTN","ASB2","MYOD1","COPS2"),group.by ='sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#CYTOPLASMIC TRANSLATION-GO
DoHeatmap(cluster.average_3hCA,features = c("RPS27A","FAU","EIF3G","RPS20","RPL38","RPS21","RPS27","RPL22L1","RPL36","RPS19","RPL36A","YBX1","RPS11","EIF2S3","RPL18A","RPS10","RPL17","RPS28","RPS6","RPL23","RPL23A","RPS16","RPS12","RPL37A","RPL3","RPL35","RPLP2","RPL9","RPL35A","RPL8","RPL27","RPL7","RPLP0","RPL19","RPL37","RPS23","RPS8","RPS24","RPS4X","UBA52","RPS14","RPS25","EIF3E","RPS13","RPL32","RPS5","RPL30","RPL29","RPL11","RPL13","RPS3","RPL28","RPS3A","RPS15A","EIF4B","RPL18","RPS18","RPL14","RPL22","RPL27A","RPL31","RPS2","EIF3D","RPS15","RPL10","RPS7","RPL6","RPL12","RPS9","RPL26","RPL15","EIF3F","RPL4","RPL10A","MIF4GD","EIF3K","RPL13A","EIF3L","EIF3H","RPL5","RPSA","RPL7A"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#KEGG-RIG 1 LIKE RECEPTOR SIGNALLING
DoHeatmap(cluster.average_3hCA,features=c("ADAR" ,"CXCL10" ,"IFIH1" ,"ISG15" ,"NFKB1" ,"TANK"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Ferroptosis-Kegg
DoHeatmap(cluster.average_3hCA,features = c("FTH1" ,"FTL" ,"GCLM" ,"SAT1" ,"SLC7A11"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#AutoImmune Thyroid Disease
DoHeatmap(cluster.average_3hCA,features=c("FAS", "GZMB", "HLA-A", "HLA-DQA1", "HLA-DQA2", "HLA-DRB1", "HLA-E", "HLA-F"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Apoptosis-kegg
DoHeatmap(cluster.average_3hCA,features = c("ATF4", "BCL2A1", "BID", "BIRC3", "CASP3", "CFLAR", "CTSC", "ENDOG", "FAS", "GZMB", "IL3RA", "NFKB1", "PMAIP1", "TNFSF10", "TRAF1"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Ribosome-KEGG
DoHeatmap(cluster.average_3hCA,features=c("MRPL23", "RPL10", "RPL10A", "RPL11", "RPL12", "RPL13", "RPL13A", "RPL14", "RPL15", "RPL17", "RPL18", "RPL18A", "RPL19", "RPL21", "RPL22", "RPL23", "RPL23A", "RPL26", "RPL27", "RPL27A", "RPL28", "RPL29", "RPL3", "RPL30", "RPL31", "RPL32", "RPL35", "RPL35A", "RPL36", "RPL37", "RPL37A", "RPL38", "RPL4", "RPL5", "RPL6", "RPL7", "RPL7A", "RPL8", "RPL9", "RPLP0", "RPLP2", "RPS10", "RPS11", "RPS12", "RPS13", "RPS14", "RPS15", "RPS15A", "RPS16", "RPS18", "RPS19", "RPS2", "RPS20", "RPS23", "RPS24", "RPS25", "RPS26", "RPS27", "RPS27A", "RPS28", "RPS3", "RPS3A", "RPS4X", "RPS4Y1", "RPS5", "RPS6", "RPS7", "RPS8", "RPS9", "RPSA", "UBA52"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Fc Epsilon RI Signaling Pathway-kegg
DoHeatmap(cluster.average_3hCA,features = c("FCER1G", "MAP2K3" ,"RAC2"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
```

Type any R code in the chunk, for example:
```{r}
#CA24h
#Subset from pDC_combined
pDC_combined_CA24h <- subset(pDC_combined, subset = treatment=='24hCA')
#Write the same code as the previous block
cluster.average_24hCA <- AverageExpression(pDC_combined_CA24h,group.by = 'subject',return.seurat = T)
#Normalize the data
cluster.average_24hCA <- NormalizeData(cluster.average_24hCA)
#Find variable genes
cluster.average_24hCA <- FindVariableFeatures(cluster.average_24hCA,selection.method = 'vst',nfeatures = 2000)
#Find PCA
cluster.average_24hCA <- RunPCA(cluster.average_24hCA)
#Find Neighbors
cluster.average_24hCA <- FindNeighbors(cluster.average_24hCA, dims = 1:20)
#Find clusters
cluster.average_24hCA <- FindNeighbors(cluster.average_24hCA, dims = 1:20)
#Run UMAP
cluster.average_24hCA <- cluster.average_3hCA <- RunUMAP(cluster.average_3hCA, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")

#Add the sex information and subject information.
#Add subject column to cluster.average_24hCA
cluster.average_24hCA@meta.data$subject <-names(Idents(cluster.average_24hCA))
# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average_24hCA@meta.data$subject, subject_sex_lookup$subject)

# Fill in the "sex" column in the cluster.average object with the corresponding values from the pDC dataset
# Find the indices of matching rows in the cluster.average object
matching_rows <- match(matching_subjects, cluster.average_24hCA@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average_24hCA@meta.data$sex <- rep(NA, nrow(cluster.average_24hCA@meta.data))

# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average_24hCA@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]
#MALE-GO
#Farensyl diphosphate metabolic process - GO
DoHeatmap(cluster.average_24hCA,features=c("FDPS","HMGCS1","FDFT1"),group.by = 'sex',disp.min = -2,disp.max = 2) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Nitric oxide biosynthetic process - GO
DoHeatmap(cluster.average_24hCA,features = c("EEF1D","EIF4A1","DNAJC3","EEF1A1","EIF3K","MYCL","PABPC1","RPLP1","RPL41","CCND3","MIF4GD","RPL24","TCF4","RPS25","RRBP1","PIM2","IL1B","IFI16","RPL18","RPS29","RPL12","JMJD1C","CCNL1","RPL10","RPLP0","FAU","RPL11","CXXC5","EEF2","NFKBIA","RPL38","RPS12","RPS27A","EIF3F","POLB","CSRNP1","RPL7A","ZFAT","RPS5","ZNF791","RBM3","RPSA","IRF4","CDKN2D","SERP1","USF2","PRKCB","RPL15","RPS21","RPL19","SP110","RPL13A","RPL31","GAS6","RPL27A","RPS9","RPL35A","RPL39","RPL36A","FOSL2","RPS3","JUND","S100A8","RPL18A","RPL21","RPL37","ARID5A","RPS11","NR4A3","RPL37A","EIF1","RPL4","RPS23","RPLP2","RPL7","EIF3L","RPS2","IER2","RPS4X","RPL23A","RPL30","S100A9","SIK1","RPS27","HES4","RPL29","RPL9","RPL35","RPL22","RPL26","RPS16","RPS28","MYBL2","RPL34","RPL27","RPS10","RPS24","VIM","RPL23","RPS7","RPS15A","RPS13","CD4","RPS6","RPS14","SPIB","RPL17","ODC1","KLF10","RPL32","SMPD3","NR4A1","RPL10A","RPL5","RPL13","EZR","GUK1","MAP2K3","APP","NR3C1","RPL3","EEF1B2","SKIL","TSC22D3","JUNB","RGCC","ARID3A","RPS8","RPS18","COMMD6","RPL6","XBP1","NR4A2","TGFB1","NCF1","RPS3A","PLAC8","ZFP36L2","IRF7","FOSB","IRF8","KDM6B","ZNF331","BTG2","PPP1R15A","RGS2","ZFP36"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Gene Expression- GO
DoHeatmap(cluster.average_24hCA,features = c("TCF4","SRSF6","RPS25","MIR142","RRBP1","PIM2","IL1B","SAMHD1","IFI16","RPL18","SRSF3","RPS29","SPON2","RPL12","JMJD1C","CCNL1","RPL10","ISCA1","RPLP0","FAU","RPL11","CXXC5","EEF2","SRGN","NFKBIA","RPL38","RPS12","RPS27A","EIF3F","POLB","CSRNP1","RPL7A","ZFAT","RPS5","ZNF791","RBM3","RPSA","SPCS1","IRF4","TNFRSF21","SERP1","USF2","LAPTM5","FBL","PRKCB","RPL15","RPS21","RPL19","MBP","SP110","RPL13A","RPL31","AHNAK","GAS6","RPL27A","ZC3HAV1","RPS9","RPL35A","GAPT","RPL39","RPL36A","FOSL2","RPS3","JUND","S100A8","RPL18A","RPL21","RPL37","ARID5A","RPS11","NR4A3","RPL37A","ISG15","SLC7A5","EIF1","RPL4","CCL3","RPS23","SRSF7","RPLP2","RPL7","EIF3L","RPS2","IER2","RPS4X","RPL23A","RPL30","S100A9","SIK1","RPS27","HES4","RPL29","RPL9","RPL35","RPL22","RPL26","RPS16","RPS28","DNAJB9","MYBL2","RPL34","RPL27","RPS10","RPS24","VIM","RPL23","TYROBP","RPS7","RPS15A","RPS13","CD4","RPS6","RPS14","SPIB","RPL17","KLF10","PTPRS","RPL32","GPX1","SERPINF1","NR4A1","RPL10A","RPL5","RPL13","EZR","MAP2K3","MIR24-2","MYADM","APP","NR3C1","WNT10A","RPL3","EEF1B2","AGPAT2","LMNA","SKIL","SNHG7","TSC22D3","ERP29","JUNB","RGCC","ARID3A","RBM38","MZB1","RPS8","RPS18","THBD","COMMD6","RPL6","XBP1","NR4A2","TGFB1","NCF1","RPS3A","PLAC8","NFKBID","ZFP36L2","CYBA","CRIP1","IRF7","FOSB","IRF8","KDM6B","PLD4","ZNF331","FCER1G","LILRA4","BTG2","PPP1R15A","RGS2","ZFP36","PTGDS"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Female Pregnanacy- GO
DoHeatmap(cluster.average_24hCA,features = c("UCP2","TIMP1","IL1B","ARHGDIB","CTSB","RPL29","JUNB","THBD","FOS","FOSB","RGS2"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#rRNA processing- GO
DoHeatmap(cluster.average_24hCA,features =c ("RPS19", "RPS15", "RPL14", "RPS25", "FBL", "RPL11", "RPS21", "RPL7A", "RPL35A", "RPS27", "RPL7", "RPS16", "RPS7", "RPS28", "RPL35", "RPL27", "RPS6", "RPS14", "RPL26", "RPS24", "RPL5", "RPL10A", "RPS8"),group.by='sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#KEGG
#Glycolysis / Gluconeogenesis-KEGG
DoHeatmap(cluster.average_24hCA,features = c("ALDH2", "ALDOA", "ENO1" ,"GAPDH" ,"LDHA" ,"PFKL" ,"PGAM1","PGK1","PKM","TPI1"),group.by='sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Trepenoid backbone biosynthesis-KEGG
DoHeatmap(cluster.average_24hCA,features=c("ACAT2","FDPS","HMGCS1", "IDI1" ,"MVD"),group.by = 'sex',disp.min = -1,disp.max = 1) +scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Covid 19 -Kegg
DoHeatmap(cluster.average_24hCA,features=c("FAU", "IL1B", "ISG15", "MX1", "MX2", "NFKBIA", "PRKCB", "RPL10", "RPL10A", "RPL11", "RPL12", "RPL13", "RPL13A", "RPL14", "RPL15", "RPL17", "RPL18", "RPL18A", "RPL19", "RPL21", "RPL22", "RPL23", "RPL23A", "RPL24", "RPL26", "RPL27", "RPL27A", "RPL28", "RPL29", "RPL3", "RPL30", "RPL31", "RPL32", "RPL34", "RPL35", "RPL35A", "RPL36A", "RPL37", "RPL37A", "RPL38", "RPL39", "RPL4", "RPL41", "RPL5", "RPL6", "RPL7", "RPL7A", "RPL9", "RPLP0", "RPLP1", "RPLP2", "RPS10", "RPS11", "RPS12", "RPS13", "RPS14", "RPS15", "RPS15A", "RPS16", "RPS18", "RPS19", "RPS2", "RPS20", "RPS21", "RPS23", "RPS24", "RPS25", "RPS27", "RPS27A", "RPS28", "RPS29", "RPS3", "RPS3A", "RPS4X","RPS4Y1","RPS5","RPS6", "RPS7" ,"RPS8", "RPS9", "RPSA"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Valine, leucine and isoleucine degradation-KEGG
DoHeatmap(cluster.average_24hCA,features=c("ACAT2", "ALDH2", "HMGCS1" ,"IL4I1"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
```

```{r}
#Subset the 3hPA data from pDC
pDC_combined_PA3h <- subset(pDC_combined, subset = treatment=='3hPA')
cluster.average_3hPA <- AverageExpression(pDC_combined_PA3h, group.by = 'subject',return.seurat = TRUE)
#Normalize the data
cluster.average_3hPA <- NormalizeData(cluster.average_3hPA)
#Find variable genes
cluster.average_3hPA <- FindVariableFeatures(cluster.average_3hPA, selection.method = "vst", nfeatures = 2000)
#Find PCA
cluster.average_3hPA <- RunPCA(cluster.average_3hPA,features =rownames(cluster.average_3hPA))
#Find Neighbors
cluster.average_3hPA <- FindNeighbors(cluster.average_3hPA, dims = 1:20)
#Find Clust
cluster.average_3hPA <- FindClusters(cluster.average_3hPA, resolution = 0.5)
#Find UMAP
cluster.average_3hPA <- RunUMAP(cluster.average_3hPA, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")
#Add subject column to cluster.average_3hhPA
cluster.average_3hPA@meta.data$subject <-names(Idents(cluster.average_3hPA))
# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average_3hPA@meta.data$subject, subject_sex_lookup$subject)

# Fill in the "sex" column in the cluster.average object with the corresponding values from the pDC dataset
# Find the indices of matching rows in the cluster.average object
matching_rows <- match(matching_subjects, cluster.average_3hPA@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average_3hPA@meta.data$sex <- rep(NA, nrow(cluster.average_3hPA@meta.data))

# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average_3hPA@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]


#regulation of lymphocyte chemotaxis
DoHeatmap(cluster.average_3hPA,features = c("CXCL10","CCL2","CCL4","CCL3"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#NK cell chemotaxix -GO
DoHeatmap(cluster.average_3hPA,features =c ("CCL2","CCL4","CCL3"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#Calcium Ion-GO
DoHeatmap(cluster.average_3hPA,features = c("CXCL10","CCL2","VMP1","CCL4","CCL3"),group.by = 'sex')

#Cytoplasmic Translation-GO
DoHeatmap(cluster.average_3hPA,features=c("SH3BGRL", "YBX1", "RPS26", "RPL35", "RPL41", "RPL9", "RPS25", "RPS5", "RPL18A", "RPL22", "RPL30", "RPS4X", "RPL11", "RPS6", "RPL21", "RPL35A", "RPL37A", "RPS3", "RPL14", "RPL34", "RPS27", "RPL13", "RPL31", "RPL32", "RPS14", "RPL27", "RPL23", "RPS15", "RPS18", "RPS9", "EIF2S3", "RPS8", "RPS13", "RPL12", "RPS15A", "RPS24", "RPL18", "RPS28", "RPS23", "EIF3D", "RPL27A", "RPL10", "RPS2", "RPL3", "RPL26", "RPS7", "RPL7", "EIF3E", "RPLP0", "RPL15", "RPL13A", "RPL23A", "RPS3A", "RPL7A", "EIF4B", "RPL10A", "RPL4", "RPL6", "RPL5", "EIF3F", "EIF3H", "EIF3L", "RPSA", "MIF4GD"),group.by='sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#Cold Induced thermogenesis-GO
DoHeatmap(cluster.average_3hPA,features =c("IRF4","CXCR4","KDM6B","UCP2","PLAC8"),group.by = 'sex',disp.min = -1,disp.max = 1)

#Dicarboxylic acid metabolic Process-GO
DoHeatmap(cluster.average_3hPA,features = c("SLC7A11","GCLM","MTHFD2","KMO"	),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Neuron differentiation-GO
DoHeatmap(cluster.average_3hPA,features = c("ITM2C", "RAC2", "CLN8", "BCL11A", "LST1", "PHACTR1", "BTG2", "IER2", "S100A9", "SERPINF1", "KLF4", "NR4A2", "SOX4", "ID1", "CXCR4", "RGS2"),group.by = 'sex',disp.min = -1,disp.max = 1) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

#Toll-like receptor signaling pathway-KEGG
DoHeatmap(cluster.average_3hPA,features = c("CCL3", "CCL4", "CXCL10", "IL1B", "NFKB1", "STAT1"),group.by='sex') + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Viral protein interaction with cytokine and cytokine receptor-KEGG
DoHeatmap(cluster.average_3hPA,features = c("CCL2", "CCL3", "CCL4", "CCR7", "CXCL10", "CXCL3", "IL2RA" ,"TNFSF10"),group.by = 'sex') + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#TNF signaling pathway-KEGG
DoHeatmap(cluster.average_3hPA,features = c("CCL2", "CCL3", "CCL4", "CCR7", "CXCL10","CXCL3", "IL2RA", "TNFSF10"),group.by='sex',disp.min = -2,disp.max = 2) + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

```
```{r}
#PA 24h
pDC_combined_PA24h <- subset(pDC_combined, subset = treatment=='24hPA')
cluster.average_24hPA <- AverageExpression(pDC_combined_PA3h, group.by = 'subject',return.seurat = TRUE)
#Normalize the data
cluster.average_24hPA <- NormalizeData(cluster.average_24hPA)
#Find variable genes
cluster.average_24hPA <- FindVariableFeatures(cluster.average_24hPA, selection.method = "vst", nfeatures = 2000)
#Find PCA
cluster.average_24hPA <- RunPCA(cluster.average_24hPA,features =rownames(cluster.average_24hPA))
#Find Neighbors
cluster.average_24hPA <- FindNeighbors(cluster.average_24hPA, dims = 1:20)
#Find Clust
cluster.average_24hPA <- FindClusters(cluster.average_24hPA, resolution = 0.5)
#Find UMAP
cluster.average_24hPA <- RunUMAP(cluster.average_24hPA, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")
#Add subject column to cluster.average_3hhPA
cluster.average_24hPA@meta.data$subject <-names(Idents(cluster.average_24hPA))
# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average_24hPA@meta.data$subject, subject_sex_lookup$subject)

# Fill in the "sex" column in the cluster.average object with the corresponding values from the pDC dataset
# Find the indices of matching rows in the cluster.average object
matching_rows <- match(matching_subjects, cluster.average_24hPA@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average_24hPA@meta.data$sex <- rep(NA, nrow(cluster.average_24hPA@meta.data))

# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average_24hPA@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]

#GO
#MHC class II protein complex-GO
DoHeatmap(cluster.average_24hPA,features = c("HLA-DQA1", "HLA-DPA1", "HLA-DRB1", "HLA-DRA", "HLA-DPB1", "HLA-DQB1", "HLA-DQA2", "HLA-DOA", "HLA-DRB5"),group.by = 'sex') + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#Sterol biosynthetic process-GO
DoHeatmap(cluster.average_24hPA,features =c("SQLE", "FDPS", "HMGCS1", "FDFT1", "MSMO1", "INSIG1", "MVD", "IDI1", "LSS", "HMGCR", "EBP", "DHCR24", "ACLY"),group.by = 'sex') + scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))
#peptide biosynthetic process-GO
DoHeatmap(cluster.average_24hPA,features=c("RPS29", "FAU", "MGST2", "EIF4B", "RPL38", "RPS15", "RPS20", "DNAJC3", "CIRBP", "RPS4Y1", "TCOF1", "RPS25", "EEF1D", "RPL12", "RPS21", "RPS5", "RPL18", "RPLP2", "RPL36A", "RPL37A", "MKNK2", "RPL7A", "EEF1A1", "RPS11", "MIF4GD", "PABPC1", "RPL37", "RPS27A", "RPLP0", "RBM3", "RPL11", "RPL23", "RPS16", "SERP1", "RPSA", "RPL27A", "RRBP1", "RPS9", "TMED2", "RPL39", "RPL13A", "RPL35A", "RPS23", "RPS10", "EIF1", "RPL10", "RPL19", "VIM", "RPL31", "RPL30", "RPS3", "RPL22", "RPL15", "RPL17", "RPL18A", "RPS28", "RPL29", "RPL27", "RPS15A", "RPL23A", "RPS2", "RPL21", "RPL35", "RPL4", "RPS27", "EEF2", "RPS7", "RPL26", "RPS4X", "RPL9", "RPS14", "EIF3F", "RPL32", "RPS6", "RPS24", "RPL7", "RPL5", "EEF1B2", "RPS13", "RPL10A", "RPL34", "RPL13", "EIF3L", "RPS8", "APP", "RPS18", "RPL3", "RPL6", "RPS3A", "ZFP36L2", "BTG2", "PPP1R15A", "RGS2", "ZFP36"),group.by = 'sex')+ scale_fill_gradientn(colors = brewer.pal(1000, "RdBu"),
                       oob = scales::oob_squish_any,
                       limits = c(-2, 2))

```
```{r}
#MTB 3h
pDC_combined_MTB3h <- subset(pDC_combined, subset = treatment=='3hMTB')
cluster.average_3hMTB <- AverageExpression(pDC_combined_MTB3h, group.by = 'subject',return.seurat = TRUE)
#Normalize the data
cluster.average_3hMTB <- NormalizeData(cluster.average_3hMTB)
#Find variable genes
cluster.average_3hMTB <- FindVariableFeatures(cluster.average_3hMTB, selection.method = "vst", nfeatures = 2000)
#Find PCA
cluster.average_3hMTB <- RunPCA(cluster.average_3hMTB,features =rownames(cluster.average_3hMTB))
#Find Neighbors
cluster.average_3hMTB <- FindNeighbors(cluster.average_3hMTB, dims = 1:20)
#Find Clust
cluster.average_3hMTB <- FindClusters(cluster.average_3hMTB, resolution = 0.5)
#Find UMAP
cluster.average_3hMTB <- RunUMAP(cluster.average_3hMTB, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")

#Add subject column to cluster.average_3hhPA
cluster.average_3hMTB@meta.data$subject <-names(Idents(cluster.average_3hMTB))
# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average_3hMTB@meta.data$subject, subject_sex_lookup$subject)

# Fill in the "sex" column in the cluster.average object with the corresponding values from the pDC dataset
# Find the indices of matching rows in the cluster.average object
matching_rows <- match(matching_subjects, cluster.average_3hMTB@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average_3hMTB@meta.data$sex <- rep(NA, nrow(cluster.average_3hMTB@meta.data))

# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average_3hMTB@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]
```
```{r}
#24h MTB

pDC_combined_MTB24h <- subset(pDC_combined, subset = treatment=='24hMTB')

cluster.average_24hMTB <- AverageExpression(pDC_combined_MTB3h, group.by = 'subject',return.seurat = TRUE)
#Normalize the data
cluster.average_24hMTB <- NormalizeData(cluster.average_24hMTB)
#Find variable genes
cluster.average_24hMTB <- FindVariableFeatures(cluster.average_24hMTB, selection.method = "vst", nfeatures = 2000)
#Find PCA
cluster.average_24hMTB <- RunPCA(cluster.average_24hMTB,features =rownames(cluster.average_24hMTB))
#Find Neighbors
cluster.average_24hMTB <- FindNeighbors(cluster.average_24hMTB, dims = 1:20)
#Find Clust
cluster.average_24hMTB <- FindClusters(cluster.average_24hMTB, resolution = 0.5)
#Find UMAP
cluster.average_24hMTB <- RunUMAP(cluster.average_24hMTB, dims = 1:20, n.neighbors = 50, min.dist = 0.25, metric = "euclidean",reduction.key = "umap",reduction.name = "umap")

#Add subject column to cluster.average_3hhPA
cluster.average_24hMTB@meta.data$subject <-names(Idents(cluster.average_24hMTB))
# Match the subject values between the two datasets
matching_subjects <- intersect(cluster.average_24hMTB@meta.data$subject, subject_sex_lookup$subject)

matching_rows <- match(matching_subjects, cluster.average_24hMTB@meta.data$subject)
# Extract the corresponding sex values from the subject_sex_lookup table
sex_values <- subject_sex_lookup[match(matching_subjects, subject_sex_lookup[["subject"]]), "sex"]
# Initialize the "sex" column in the cluster.average object with NAs
cluster.average_24hMTB@meta.data$sex <- rep(NA, nrow(cluster.average_24hMTB@meta.data))

# Assign the sex values to matching rows in the "sex" column of cluster.average
cluster.average_24hMTB@meta.data$sex[!is.na(matching_rows)] <- sex_values[!is.na(matching_rows)]
```
```{r}
#Comparing GO between UT, 3hCA, 24hCA
#First pathway
#Nitric Oxide Metabolic Process
heatmap_NO_UT <- DoHeatmap(cluster.average_UT, features = c("TSPO", "KLF4", "HSP90AA1", "CD36", "IL1B", "AIF1", "SOD2"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_NO_UT
heatmap_NO_3hCA <- DoHeatmap(cluster.average_3hCA, features = c("TSPO", "KLF4", "HSP90AA1", "CD36", "IL1B", "AIF1", "SOD2"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_NO_24hCA <- DoHeatmap(cluster.average_24hCA, features = c("TSPO", "KLF4", "HSP90AA1", "CD36", "IL1B", "AIF1", "SOD2"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_NO_24hCA
grid.arrange(heatmap_NO_UT, heatmap_NO_3hCA, heatmap_NO_24hCA, ncol = 3)
```
```{r}
#Carbon Metabolic Process
heatmap_C_UT <- DoHeatmap(cluster.average_UT, features = c("GRSF1","OCIAD1","RAB21","SOCS1","PTGDS","CSF2RB","MARCH1"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_C_3hCA <- DoHeatmap(cluster.average_3hCA, features = c("GRSF1","OCIAD1","RAB21","SOCS1","PTGDS","CSF2RB","MARCH1"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_C_24hCA <- DoHeatmap(cluster.average_24hCA, features = c("GRSF1","OCIAD1","RAB21","SOCS1","PTGDS","CSF2RB","MARCH1"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_C_UT, heatmap_C_3hCA, heatmap_C_24hCA, ncol = 3)

```
```{r}
#Glycolysis
heatmap_Gly_UT <- DoHeatmap(cluster.average_UT, features = c("GAPDH","ADH5","ADPGK","PGK1","LDHA","ALDH3B1","MINPP1","PGAM1","PFKM","PFKL","ALDH9A1","ALDH2","PKM","GPI","ALDH7A1","PGAM4","FBP1","ALDOA","ENO2","LDHB","ALDH1B1","HK3"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Gly_3hCA <- DoHeatmap(cluster.average_3hCA, features = c("GAPDH","ADH5","ADPGK","PGK1","LDHA","ALDH3B1","MINPP1","PGAM1","PFKM","PFKL","ALDH9A1","ALDH2","PKM","GPI","ALDH7A1","PGAM4","FBP1","ALDOA","ENO2","LDHB","ALDH1B1","HK3"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Gly_24hCA <- DoHeatmap(cluster.average_24hCA, features = c("GAPDH","ADH5","ADPGK","PGK1","LDHA","ALDH3B1","MINPP1","PGAM1","PFKM","PFKL","ALDH9A1","ALDH2","PKM","GPI","ALDH7A1","PGAM4","FBP1","ALDOA","ENO2","LDHB","ALDH1B1","HK3"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Gly_UT, heatmap_Gly_3hCA, heatmap_Gly_24hCA, ncol = 3)

```
```{r}
#TCA Cycle

heatmap_TCA_UT <- DoHeatmap(cluster.average_UT, features = c("PDHB","MDH2","OGDH","SDHC","FH","DLAT","ACO2","SDHA","IDH3G","ACLY","CS","SUCLG1","DLD","SDHD","IDH3A"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_TCA_3hCA <- DoHeatmap(cluster.average_3hCA, features = c("PDHB","MDH2","OGDH","SDHC","FH","DLAT","ACO2","SDHA","IDH3G","ACLY","CS","SUCLG1","DLD","SDHD","IDH3A"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_TCA_24hCA <- DoHeatmap(cluster.average_24hCA, features = c("PDHB","MDH2","OGDH","SDHC","FH","DLAT","ACO2","SDHA","IDH3G","ACLY","CS","SUCLG1","DLD","SDHD","IDH3A"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_TCA_UT, heatmap_TCA_3hCA, heatmap_TCA_24hCA, ncol = 3)
```
```{r}
#Oxidative Phosphorylation

heatmap_OP_UT <- DoHeatmap(cluster.average_UT, features = c("NDUFA11","COX5B","ATP5ME","COX8A","UQCRQ","NDUFB1","NDUFA1","ATP5MC2","NDUFS6","COX7C","UQCR11","COX6B1","NDUFB3","ATP5MG","UQCRH","COX7B2"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_OP_3hCA <- DoHeatmap(cluster.average_3hCA, features = c("NDUFA11","COX5B","ATP5ME","COX8A","UQCRQ","NDUFB1","NDUFA1","ATP5MC2","NDUFS6","COX7C","UQCR11","COX6B1","NDUFB3","ATP5MG","UQCRH","COX7B2"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_OP_24hCA <- DoHeatmap(cluster.average_24hCA, features = c("NDUFA11","COX5B","ATP5ME","COX8A","UQCRQ","NDUFB1","NDUFA1","ATP5MC2","NDUFS6","COX7C","UQCR11","COX6B1","NDUFB3","ATP5MG","UQCRH","COX7B2"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_OP_UT, heatmap_OP_3hCA, heatmap_OP_24hCA, ncol = 3)
```
```{r}
#Farnesyl Diphosphate Metabolic Process
heatmap_FDP_UT <- DoHeatmap(cluster.average_UT, features = c("FDPS","HMGCS1","FDFT1"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_FDP_3hCA <- DoHeatmap(cluster.average_3hCA, features = c("FDPS","HMGCS1","FDFT1"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_FDP_24hCA <- DoHeatmap(cluster.average_24hCA, features = c("FDPS","HMGCS1","FDFT1"), group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_FDP_UT, heatmap_FDP_3hCA, heatmap_FDP_24hCA, ncol = 3)
```
```{r}
#Cellular Nitrogen Compound Biosynthetic Process
genes<-c("RPS4Y1", "TCOF1", "DDX5", "BCL11A", "RPS25", "EEF1D", "RPL12", "S100A9", "RPS21", "ELF2", "RPS5", "RPL18", "RPLP2", "RPL36A", "RPL37A", "RBCK1", "IRF1", "MKNK2", "RPL7A", "EEF1A1", "ADA", "JMJD1C", "PRKCB", "RPS11", "MIF4GD", "PABPC1", "ZFAT", "RPL37", "BAZ1A", "RPS27A", "RPLP0", "CLN8", "RBM3", "NADK", "S100A8", "RPL11", "PIM2", "CSRNP1", "RPL23", "IFI16", "RPS16", "TRIM22", "USF2", "SERP1", "RPSA", "RPL27A", "RRBP1", "RPS9", "TMED2", "ZNF791", "RPL39", "RPL13A", "RPL35A", "RPS23", "POLB", "RPS10", "GAS6", "EIF1", "NR4A3", "RPL10", "RPL19", "VIM", "CCNL1", "RPL31", "RPL30", "RPS3", "RPL22", "RPL15", "SP110", "JUND", "CDKN2D", "RPL17", "RPL18A", "RPS28", "DCK", "RPL29", "ARID5A", "SPIB", "RPL27", "IRF4", "RPS15A", "FOSL2", "RPL23A", "RPS2", "RPL21", "RPL35", "SIK1", "CD4", "RPL4", "RPS27", "EEF2", "RPS7", "RPL26", "NFKBIA", "RPS4X", "RPL9", "CXXC5", "ODC1", "IER2", "RPS14", "EIF3F", "RPL32", "RPS6", "RPS24", "RPL7", "RPL5", "HES4", "EEF1B2", "RPS13", "RPL10A", "RPL34", "RPL13", "SMPD3", "NR3C1", "EZR", "KLF10", "MYBL2", "SKIL", "NR4A1", "COMMD6", "EIF3L", "TSC22D3", "RPS8", "MAP2K3", "APP", "RGCC", "RPS18", "GUK1", "RPL3", "JUNB", "RPL6", "TGFB1", "ARID3A", "NR4A2", "XBP1", "RPS3A", "ZFP36L2", "NCF1", "FOSB", "ZNF331", "BTG2", "KDM6B", "PPP1R15A", "IRF7","IRF8","RGS2","PLAC8","ZFP36")
heatmap_NCBUT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_NCB3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_NCB24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_NCBUT, heatmap_NCB3hCA, heatmap_NCB24hCA, ncol = 3)
```
```{r}
#Cellular amide metabolic process
genes<-c("RPS4Y1","TCOF1","RPS25","EEF1D","RPL12","TAPBP","RPS21","RPS5","RPL18","RPLP2","RPL36A","RPL37A","MKNK2","RPL7A","EEF1A1","ADA","RPS11","MIF4GD","PABPC1","RPL37","RPS27A","RPLP0","CLN8","RBM3","RPL11","RPL23","RPS16","SPCS1","SERP1","RPSA","RPL27A","RRBP1","RPS9","TMED2","RPL39","RPL13A","RPL35A","RPS23","RPS10","EIF1","RPL10","RPL19","VIM","RPL31","RPL30","RPS3","RPL22","RPL15","RPL17","RPL18A","RPS28","RPL29","RPL27","RPS15A","RPL23A","RPS2","RPL21","RPL35","RPL4","RPS27","EEF2","RPS7","RPL26","RPS4X","RPL9","RPS14","EIF3F","RPL32","RPS6","RPS24","RPL7","RPL5","EEF1B2","RPS13","RPL10A","RPL34","RPL13","SMPD3","EIF3L","RPS8","APP","RPS18","RPL3","RPL6","RPS3A","ZFP36L2","BTG2","PPP1R15A","RGS2","ZFP36")

heatmap_AM_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_AM_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_AM_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_AM_UT, heatmap_AM_3hCA, heatmap_AM_24hCA, ncol = 3)
```
```{r}
#Peptide Metabolic Process
genes <- c("RPS29","FAU","MGST2","EIF4B","RPL38","RPS15","RPS20","DNAJC3","CIRBP","RPS4Y1","TCOF1","RPS25","EEF1D","RPL12","TAPBP","RPS21","RPS5","RPL18","RPLP2","RPL36A","RPL37A","MKNK2","RPL7A","EEF1A1","RPS11","MIF4GD","PABPC1","RPL37","RPS27A","RPLP0","RBM3","RPL11","RPL23","RPS16","SPCS1","SERP1","RPSA","RPL27A","RRBP1","RPS9","TMED2","RPL39","RPL13A","RPL35A","RPS23","RPS10","EIF1","RPL10","RPL19","VIM","RPL31","RPL30","RPS3","RPL22","RPL15","RPL17","RPL18A","RPS28","RPL29","RPL27","RPS15A","RPL23A","RPS2","RPL21","RPL35","RPL4","RPS27","EEF2","RPS7","RPL26","RPS4X","RPL9","RPS14","EIF3F","RPL32","RPS6","RPS24","RPL7","RPL5","EEF1B2","RPS13","RPL10A","RPL34","RPL13","EIF3L","RPS8","APP","RPS18","RPL3","RPL6","RPS3A","ZFP36L2","BTG2","PPP1R15A","RGS2","ZFP36")
heatmap_Pep_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Pep_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Pep_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Pep_UT, heatmap_Pep_3hCA, heatmap_Pep_24hCA, ncol = 3)
```
```{r}
#rRNA metabolic process
genes <- c("RPL14", "GTF3A", "RPS15", "TCOF1", "RPS25", "RPS21", "RPL7A", "RPL11", "FBL", "RPS16", "RPL35A", "RPS28", "RPL27", "RPL35", "RPS27", "RPS7", "RPL26", "ISG20", "RPS14", "RPS6", "RPS24", "RPL7", "RPL5", "RPL10A", "RPS8")
heatmap_rRNA_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_rRNA_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_rRNA_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_rRNA_UT, heatmap_rRNA_3hCA, heatmap_rRNA_24hCA, ncol = 3)

```
```{r}
#Apoptotic process
genes <- c("ADAR", "SLC25A6", "POLB", "SIVA1", "GAS6", "TPT1", "RPL10", "TNFRSF21", "CCNL1", "RPS3", "CDKN2D", "CTSB", "PMAIP1", "PIM3", "SIK1", "RPS7", "RPL26", "NFKBIA", "CCL3", "TRAF4", "LMNA", "RNF130", "RPS6", "UCP2", "B4GALT1", "NR3C1", "MYBL2", "SKIL", "NR4A1", "GADD45B", "CXCR4", "TSC22D3", "HSP90B1", "APP", "RGCC", "DNASE1L3", "TGFB1", "ERP29", "NR4A2", "MCL1", "XBP1", "IFIT2", "HERPUD1", "RPS3A", "MX1", "ITM2C", "NCF1", "MZB1", "CRIP1", "TCL1A", "BTG2", "PTCRA", "PPP1R15A", "IRF7", "PLAC8", "ZFP36")

heatmap_apop_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_apop_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_apop_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_apop_UT, heatmap_apop_3hCA, heatmap_apop_24hCA, ncol = 3)
```
```{r}
#Ferroptosis
genes <- c("FTH1", "FTL", "GCLM" ,"SAT1", "SLC7A11")
heatmap_ferrop_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_ferrop_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_ferrop_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_ferrop_UT, heatmap_ferrop_3hCA, heatmap_ferrop_24hCA, ncol = 3)
```
```{r}
#Carboxylic acid metabolic process
genes <- c("PKM", "GSTP1", "ENO1", "TPI1", "GAPDH", "GPX4", "ACAT2", "IL4I1", "SCD", "MSMO1", "INSIG1", "LDHA", "MTHFD2", "DBI", "ALDOA", "ADTRP", "KYNU", "PGD", "FADS2", "IDH2", "PGAM1", "VDAC1", "FADS1", "PGK1", "ELOVL5", "PFKL", "ACLY", "STARD4", "ACSL4", "ARL2")

heatmap_Caboxy_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Caboxy_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Caboxy_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Caboxy_UT, heatmap_Caboxy_3hCA, heatmap_Caboxy_24hCA, ncol = 3)
```
```{r}
#Cholestrol biosynthetic process
genes <- c("FDPS","HMGCS1","FDFT1","MSMO1","INSIG1","MVD","IDI1","LSS","HMGCR","EBP","DHCR24","ACLY")
heatmap_Chole_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))
heatmap_Chole_UT
heatmap_Chole_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Chole_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))
heatmap_Chole_24hCA
grid.arrange(heatmap_Chole_UT, heatmap_Chole_3hCA, heatmap_Chole_24hCA, ncol = 3)

```
```{r}
#Cell Migration-GO
genes <- c("SMPD3", "CCL5", "AIF1", "B4GALT1", "CORO1A", "MAP2K3", "RAC2", "SELPLG", "PHACTR1", "SERPINF1", "GPX1", "LDLRAD4", "DUSP1", "PPBP", "PYCARD", "ACTG1", "KLF4", "ARHGDIB", "S100A8", "NR4A2", "NR4A1", "S100A9", "MYADM", "ID1", "RGCC", "CXCR4", "FCER1G")

heatmap_Migra_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Migra_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Migra_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Migra_UT, heatmap_Migra_3hCA, heatmap_Migra_24hCA, ncol = 3)
```
```{r}
#Valine, leucine and isoleucine degradation
genes <- c("ACAT2", "ALDH2", "HMGCS1" ,"IL4I1")
heatmap_Stero_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Stero_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Stero_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Stero_UT, heatmap_Stero_3hCA, heatmap_Stero_24hCA, ncol = 3)
```
```{r}
#Steroid biosynthesis Pathway
genes <- c("SQLE","FDPS","HMGCS1","FDFT1","MSMO1","INSIG1","NFKB1","MVD","IDI1","LSS","HMGCR","EBP","DHCR24","ACLY","STARD4")
heatmap_Stero_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Stero_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Stero_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Stero_UT, heatmap_Stero_3hCA, heatmap_Stero_24hCA, ncol = 3)
```
```{r}
#Calcium signalling Pathway
genes <- c("CALM2", "CXCR4", "GNA15" ,"ORAI2" ,"PRKCB" ,"SLC25A5" ,"SLC25A6", "VEGFB")
heatmap_Cal_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Cal_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Cal_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(200, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Cal_UT, heatmap_Cal_3hCA, heatmap_Cal_24hCA, ncol = 3)
```
```{r}
#Negative Regulation of Immune System Process-GO
genes <- c("IL2RA","IL7R","ISG15","AHR","IFI16","PARP14","IL4I1","ADTRP","LPXN","PAG1","CCL3","HLA-A","ADAR","IRF1","HLA-F","RUNX3","CD47","USP18","OAS1","HLA-E","CASP3","CCL2","HLA-DRB1","SERPINB9","SAMSN1","BTN2A2","NMI","TNFAIP3","ELF1")
heatmap_Negati_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Negati_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Negati_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Negati_UT, heatmap_Negati_3hCA, heatmap_Negati_24hCA, ncol = 3)

```
```{r}
#Defense Response to Virus-GO
genes <- c("CXCL10","ISG15","PMAIP1","BIRC3","ISG20","IFIT2","MX1","IFIT3","IFI44L","IFI16","PLSCR1","STAT1","OAS2","EIF2AK2","NT5C3A","IFIH1","ADAR","HERC5","IRF1","IFI6","TANK","OAS1","MX2")

heatmap_Def_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Def_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Def_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Def_UT, heatmap_Def_3hCA, heatmap_Def_24hCA, ncol = 3)

```
```{r}
#Regulation of Lympocyte Chemotaaxis-GO
genes <- c("CXCL10","CCL2","CCL4","CCL3")

heatmap_Lym_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Lym_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Lym_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Lym_UT, heatmap_Lym_3hCA, heatmap_Lym_24hCA, ncol = 3)
```
```{r}
#IL-6 production-GO
genes <- c("ARRB2","TNFAIP3","PYCARD","TYROBP","CD36","IL1B","AIF1")
heatmap_IL6_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_IL6_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_IL6_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_IL6_UT, heatmap_IL6_3hCA, heatmap_IL6_24hCA, ncol = 3)
```
```{r}
#IL-17 production-GO
genes <-c("RFTN1","SLC7A5","IFNG" ,"LILRB2", "AIF1",  "AKIRIN2" ,"CD36", "EREG", "ZBTB20", "HYAL2" ,"APP" ,"TSLP" ,"F2R")
heatmap_IL17_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_IL17_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_IL17_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_IL17_UT, heatmap_IL17_3hCA, heatmap_IL17_24hCA, ncol = 3)
```
```{r}
#Cell death-GO
genes <- c("CXCL10","OPN3","CTSC","TXN","BCL2A1","IL2RA","ENDOG","IL7R","CFLAR","BTG1","GCLM","SLC7A11","NFKB1","CDKN1A","PMAIP1","BIRC3","IFIT2","MX1","IFIT3","SQSTM1","AHR","IFI16","PLSCR1","TNFSF10","PTPN1","BID","ATF5","STAT1","TNFAIP8","EIF2AK2","AKR1C3","KMO","MARCKS","CCR7","ARL6IP5","GZMB","CCL3","NINJ1","SRGN","XAF1","STK4","ADAR","EMP3","IRF1","IFI6","HLA-F","TAF9","CASP3","TRAF1","PKM","SP100","DNAJA1","CCL2","CD38","ACTN4","FAS","NUP62","ATF4","SERPINB9","RTN4","RBCK1","HSP90B1")
heatmap_death__UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_death__3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_death__24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_death__UT, heatmap_death__3hCA, heatmap_death__24hCA, ncol = 3)
```
```{r}
#Auto immune thyroid desease-GO
genes <- c("FAS", "GZMB", "HLA-A", "HLA-DQA1", "HLA-DQA2", "HLA-DRB1", "HLA-E", "HLA-F")
heatmap_auto_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_auto_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_auto_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_auto_UT, heatmap_auto_3hCA, heatmap_auto_24hCA, ncol = 3)
```
```{r}
#MHC class II peptide complex assembly-GO
genes <- c("HLA-DQA1","HLA-DPA1","HLA-DRB1","HLA-DRA","HLA-DPB1","HLA-DQB1","HLA-DQA2","HLA-DOA","HLA-DRB5")
heatmap_MHC_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_MHC_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_MHC_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_MHC_UT, heatmap_MHC_3hCA, heatmap_MHC_24hCA, ncol = 3)
```
```{r}
#T cell mediated immunity-GO
genes <- c("IL7R","MYO1G","IL4I1","CD70","RFTN1","HLA-DRB1","HLA-DRA","CTSH","HLA-A","MALT1")
heatmap_T_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_T_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_T_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_T_UT, heatmap_T_3hCA, heatmap_T_24hCA, ncol = 3)
```
```{r}
#RIG -I like receptor signaling pathway-GO
genes <- c("ADAR", "CXCL10", "IFIH1", "ISG15", "NFKB1", "TANK")
heatmap_Rig_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Rig_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Rig_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Rig_UT, heatmap_Rig_3hCA, heatmap_Rig_24hCA, ncol = 3)
```
```{r}
#NOD like receptor signaling pathway-GO
genes <- c("BIRC3", "CCL2" ,"CXCL3" ,"IFI16" ,"IL1B" ,"NFKB1", "OAS2", "STAT1", "TXN")
heatmap_NOD_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_NOD_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_NOD_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_NOD_UT, heatmap_NOD_3hCA, heatmap_NOD_24hCA, ncol = 3)
```
```{r}
#Necropotosis-GO
genes <- c("BID", "BIRC3", "CFLAR", "EIF2AK2", "FTH1", "FTL", "IL1B", "SQSTM1", "STAT1", "TNFSF10")
heatmap_Necro_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Necro_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Necro_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Necro_UT, heatmap_Necro_3hCA, heatmap_Necro_24hCA, ncol = 3)
```
```{r}
#Cytokine-Cytokine receptor Interaction
genes <-c("CCL2","CCL3","CCL4","CCR7","CXCL10","CXCL3","IL1B","IL1RN","IL2RA","IL7R","TNFRSF4","TNFSF10")
heatmap_cytocyto_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_cytocyto_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_cytocyto_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_cytocyto_UT, heatmap_cytocyto_3hCA, heatmap_cytocyto_24hCA, ncol = 3)
```
```{r}
#Intestinal immune network for IgA production
genes <-c("BCL6","CD79A","CD79B","CD83","CXCL12","CXCL13","CXCR4","IL10","IL21","IL21R","IL6","LTB","LTBR","TNFRSF13B")
heatmap_intestine_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_intestine_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_intestine_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_intestine_UT, heatmap_intestine_3hCA, heatmap_intestine_24hCA, ncol = 3)
```
```{r}
#Ribosome
genes <-c("MRPL23", "RPL10", "RPL10A", "RPL11", "RPL12", "RPL13", "RPL13A", "RPL14", "RPL15", "RPL17", "RPL18", "RPL18A", "RPL19", "RPL21", "RPL22", "RPL23", "RPL23A", "RPL26", "RPL27", "RPL27A", "RPL28", "RPL29", "RPL3", "RPL30", "RPL31", "RPL32", "RPL35", "RPL35A", "RPL36", "RPL37", "RPL37A", "RPL38", "RPL4", "RPL5", "RPL6", "RPL7", "RPL7A", "RPL8", "RPL9", "RPLP0", "RPLP2", "RPS10", "RPS11", "RPS12", "RPS13", "RPS14", "RPS15", "RPS15A", "RPS16", "RPS18", "RPS19", "RPS2", "RPS20", "RPS23", "RPS24", "RPS25", "RPS26", "RPS27", "RPS27A", "RPS28", "RPS3", "RPS3A", "RPS4X", "RPS4Y1", "RPS5", "RPS6", "RPS7", "RPS8", "RPS9", "RPSA", "UBA52")

heatmap_Ribosome_UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Ribosome_3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Ribosome_24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Ribosome_UT, heatmap_Ribosome_3hCA, heatmap_Ribosome_24hCA, ncol = 3)
```
```{r}
#Visualizing the common pathways in the three time points
#response to cytokine-GO
source("Gene_Formatting.R")
genes_24h <- c("BCLAF1", "SIRPA", "IL10RB", "IL10RA", "CREB1", "PRPF8", "TMSB4X", "SPI1", "LGALS9", "NDUFA13", "SH2B3", "ZFP36", "TAF9", "KIF5B", "TNFRSF18", "ENDOG", "MBP", "CD44", "CXCL1", "TPR", "ETV3", "CXCL3", "PYCARD", "CD40", "CD58", "UBE2G2", "TFRC", "PTPN2", "BIRC3", "PARP9", "RAD23B", "ACTR3", "TRIM44", "RNMT", "HCLS1", "IL13RA1", "TP53", "CYLD", "LAMP3", "CCL22", "PTPRC", "IFI16", "TANK", "ZFP36L1", "CSF2RA", "IFNGR1", "HCK", "IL7R", "CDC42", "TNFRSF4", "NRP2", "IFNAR1", "SETD2", "LAPTM5", "HIF1A", "EBI3", "GSN", "CSF2RB", "SAMHD1", "CD70")

genes_3h <- format_gene_list("CXCL10/CCL3/ISG15/IFIT2/IFIT3/IFIH1/CXCL2/MX1/OAS1/IL1B/SOCS1/RNMT/USP18/TNFSF13B/XAF1/PARP14/ZFP36/IRF1/EIF2AK2/MX2/NMI/LGALS9/XRCC5/XBP1/PNPT1/SP100/B3GNT2/MIR142/IL3RA/TNFRSF4/PTPRC/GCLM/FCER1G/STAT3/BST2/CD38/ENDOG/NDUFA13/TRIM44/IFI16/CCL4/MTF2/LILRA4/MYD88/IRF8/ADAR/APPL1/NUB1/PML/DHX9")

genes_UT <- format_gene_list("CXCL10/IL2RA/ENDOG/IL7R/ISG15/GCLM/NFKB1/TNFRSF4/BIRC3/IFIT2/MX1/IFIT3/CCL4/IFI16/PLSCR1/PTPN1/STAT1/PARP14/NFAT5/OAS2/EIF2AK2/KMO/IFIH1/CCR7/CCL3/XAF1/B3GNT2/ADAR/RBMX/IRF1/TAF9/TANK/CD47/USP18/OAS1/MX2/CASP3/TRAF1/SP100/CCL2/CD38/ACTN4/FAS")

#Select the unique genes from the three time points
genes <- unique(c(genes_24h, genes_3h, genes_UT))

heatmap_Cytokine__UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Cytokine__3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Cytokine__24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Cytokine__UT, heatmap_Cytokine__3hCA, heatmap_Cytokine__24hCA, ncol = 3)

```
```{r}
#Negative Regulation of Immune system process-GO

genes_UT <- format_gene_list("IL2RA/IL7R/ISG15/AHR/IFI16/PARP14/IL4I1/ADTRP/LPXN/PAG1/CCL3/HLA-A/ADAR/IRF1/HLA-F/RUNX3/CD47/USP18/OAS1/HLA-E/CASP3/CCL2/HLA-DRB1/SERPINB9/SAMSN1/BTN2A2/NMI/TNFAIP3/ELF1")

genes_3h <- format_gene_list("CCL3/ISG15/OAS1/SOCS1/USP18/PARP14/IRF1/HLA-E/NMI/LGALS9/CD55/HLA-A/HLA-B/PTPRC/BST2/BTN2A2/IFI16/IL4I1/ADAR/APPL1")

genes_24h <- format_gene_list("CD46/PTEN/BTN2A2/TNFAIP3/PARP1/RUNX3/CUEDC2/TYROBP/PRKAR1A/ARRB2/USP15/TRIB1/SPI1/LGALS9/CNN2/DUSP3/SPN/ID2/HLA-DRB1/AHR/PTPN2/PTGER4/TMBIM6/CD86/PTPRC/IFI16/GPX1/HCK/IL7R/IL4I1/CD55/RASSF5/LAPTM5/HLA-DOA/CST7/SAMHD1")

genes <- unique(c(genes_24h, genes_3h, genes_UT))

heatmap_Negative_Immune__UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Negative_Immune__3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Negative_Immune__24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Negative_Immune__UT, heatmap_Negative_Immune__3hCA, heatmap_Negative_Immune__24hCA, ncol = 3)
```
```{r}
#[3] "regulation of cold-induced thermogenesis"
genes_UT <- format_gene_list("IRF4/KDM6B/UCP2/CXCR4/PLAC8")

genes_3h <- format_gene_list("ADIPOR1/ATF4/RHEB/NR1H2/CXCR4/UCP2/FABP5/KDM6B/IRF4")

genes_24h <- format_gene_list("PLAC8/IRF4/FABP5/CXCR4/GNAS/DYNC1H1")

genes <- unique(c(genes_24h, genes_3h, genes_UT))

heatmap_Cold__UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Cold__3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_Cold__24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_Cold__UT, heatmap_Cold__3hCA, heatmap_Cold__24hCA, ncol = 3)
```
```{r}
#regulation of immune system process
genes_UT<- format_gene_list("CXCL10/CTSC/IL2RA/IL7R/ISG15/SLC7A11/CDKN1A/TNFRSF4/BIRC3/AHR/CCL4/IFI16/PLSCR1/PTPN1/BID/STAT1/PARP14/IL4I1/ADTRP/MYO1G/CD83/LPXN/IFIH1/PAG1/CCR7/CCL3/NINJ1/HLA-A/ADAR/IRF1/HLA-F/RUNX3/CD47/USP18/OAS1/HLA-E/CASP3/HLA-DQA1/CCL2/CD38/HLA-DRB1/SERPINB9/RTN4/RBCK1/SAMSN1/HLA-DQA2/ZNFX1/SLC15A4/BTN2A2/NMI/TNFAIP3/RFTN1")

genes_3h <- format_gene_list("CXCL10/CCL3/ISG15/IFIH1/CTSC/OAS1/IL1B/SOCS1/USP18/SLC15A4/TNFSF13B/RBCK1/PARP14/ZFP36/IRF1/HLA-E/NMI/LGALS9/XRCC5/XBP1/IFI35/CD55/MIR142/LGMN/HLA-A/HLA-B/ARID5A/PQBP1/TNFRSF4/PTPRC/NFKBIZ/FCER1G/STAT3/CDC73/BST2/BTN2A2/CD38/JUNB/IFI16/CCL4/CD99/CALR/IL4I1/LILRA4/MYD88/ADAR/APPL1/CDKN1A/B2M/KHDRBS1/ZNFX1/DHX9/TAPBP/SMARCC1/IRF7/ELF1/HLA-F/MSN")

genes_24h <- format_gene_list("PRKDC/NOTCH2/PQBP1/RPS6KA3/HLA-DPA1/CD46/LGALS1/PTEN/BTN2A2/HLA-DPB1/SUPT6H/SH3KBP1/TNFAIP3/PARP1/HLA-DQB1/HSPH1/PRMT1/RUNX3/CUEDC2/HAX1/TYROBP/PRKAR1A/PLSCR1/NR4A3/CALR/ARRB2/SIRPA/USP15/TRIB1/RB1/GPSM3/JUNB/CDKN1A/CREB1/RASSF2/SPI1/CLPTM1/LGALS9/CNN2/SWAP70/BID/DHPS/ZFP36/DUSP3/TNFRSF18/NPLOC4/PPP2R3C/PHB2/EP300/PYCARD/HLA-DMB/EIF6/SPN/CD40/ID2/HLA-DRB1/TFRC/AHR/PTPN2/BIRC3/PARP9/CNOT4/PTGER4/ACTB/PJA2/TMBIM6/CD86/HCLS1/CYLD/PTPRC/SASH3/IFI16/ZFP36L1/GPX1/CTSH/HCK/ICOSLG/UBE2N/HSP90AA1/IL7R/LY96/HLA-DQA2/IL4I1/CD83/CD55/BAX/TNFRSF4/RASSF5/LAPTM5/DDX3X/HIF1A/EBI3/HLA-DOA/CST7/SAMHD1/CD70")


genes <- unique(c(genes_24h, genes_3h, genes_UT))

heatmap_immune_reg__UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_immune_reg__3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_immune_reg__24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_immune_reg__UT, heatmap_immune_reg__3hCA, heatmap_immune_reg__24hCA, ncol = 3)
```
```{r}
#CA-
#1-cell surface receptor signaling pathway"

genes_UT <- format_gene_list("ISG15/AGPAT2/MX1/SEMA7A/CSF2RB/NFKBID/TNFRSF4/CCL3/PLSCR1/ADAR/WNT10A/SOCS1/IRF1/SKIL/CD47/EIF5A/PLEK/ATP6AP2/P2RY6/IL6ST/UGCG/RBCK1/STK4/SULF2/YTHDF2/HLA-A/ARHGDIA/HLA-DQB1/RFTN1/LPXN/BIRC3/HSPA5/TRIM44/IL10RA/ICOSLG/NR4A2/ARID4B/BAIAP2/PMAIP1/TNFSF13B/GAS6/RBM4/CDC42/RNF213/STAT6/RAP1A/IRF9/P2RY14/APH1A/LEPROTL1/MAD2L2/NFKBIZ/DGKZ/PMEPA1/STAT1/NR4A3/RB1CC1/CSF2RA/BIRC2/VPS35/DDX5/HCLS1/SUB1/BID/LILRB4/NFKB1/SMPD3/SP100/ZFAND5/SNX6/PTPN2/DDB1")

genes_3h <- format_gene_list("CXCL10/CCL3/ISG15/TNFSF10/IFI6/CXCL2/MX1/ATF3/OAS1/IL1B/SOCS1/RNF213/USP18/TNFSF13B/RBCK1/PARP14/IRF1/PILRB/NMI/SP100/CD55/LGMN/HLA-A/IL3RA/SUB1/IFT57/BAG1/TNFRSF4/USP47/PTPRC/GCLM/NFKBIZ/FCER1G/STAT3/STMN1/CDC73/BTN2A2/CD38/OCIAD1/SUDS3/USP34/NDUFA13/TRIM44/CCL4")

genes_24h <- format_gene_list("TMSB4X/RASSF2/SPI1/NDUFA13/CSNK1D/BID/TRIM33/SH2B3/PFDN5/SIAH2/AKT1S1/OSBPL8/DUSP3/CPEB4/TNFRSF18/CSNK1A1/CD44/CXCL1/PHB2/TCTN3/CXCL3/BAIAP2/EP300/PYCARD/USP34/SPN/CD40/HLA-DRB1/ECE1/PTPN2/RAB14/BIRC3/PARP9/MBD2/MTMR4/PSME3/RAB7A/PDCD6/TRIM44/CD86/HCLS1/IL13RA1/SPRED2/TP53/HSPB1/CYLD/CCL22/PTPRC/TANK/PEA15/CSF2RA/GPX1/IFNGR1/HCK/ICOSLG/UBE2N/IL7R/HIVEP1/LY96/CDC42/CD55/BAX/VPS35/TNFRSF4/NRP2/BCL2A1/IFNAR1/CCNC/CALCRL/LAPTM5/MTSS1/DDX3X/HIF1A/ARHGDIA/EBI3/CSF2RB/PTGIR/SAMHD1/CD70")

genes <- unique(c(genes_24h, genes_3h, genes_UT))

heatmap_cell_sur__UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_cell_sur__3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_cell_sur__24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_cell_sur__UT, heatmap_cell_sur__3hCA, heatmap_cell_sur__24hCA, ncol = 3)

```
```{r}
#2 Immune Response
genes_UT <- format_gene_list("ISG15/PTGDS/IFIT2/IFI44L/MX1/CTSC/SEMA7A/CSF2RB/NFKBID/ISG20/GZMB/NINJ1/TNFRSF4/CCL3/HLA-DQA1/PLSCR1/ADAR/TAP1/SOCS1/IRF1/CD47/HLA-F/IL6ST/LTB/RBCK1/YTHDF2/GPR183/HLA-A/HLA-DQB1/RFTN1/LPXN/BIRC3/ICOSLG/XRCC5/TNFSF13B/REL/SLAMF7/PSMA1/CDC42/HLA-B/STAT6/PPP1R14B/TRIM22/MAD2L2/NFKBIZ/CLEC4C/DGKZ/MX2/STAT1/NR4A3/ERCC1/MPEG1/ACTR3/BIRC2/PSMB10/LITAF/TNIP1/LILRB4/SLC7A5/TRAF4/SP100/TAPBP/NFKB2/IFI16/PTPN2")

genes_3h <- format_gene_list("CXCL10/CCL3/ISG15/IFIT2/IFIT3/TNFSF10/LTB/IFIH1/HERC5/CTSC/IFI6/CXCL2/MX1/RGS1/OAS1/IL1B/SOCS1/IFI44L/USP18/SLC15A4/NKG7/TNFSF13B/RBCK1/EXOSC9/PARP14/IRF1/EIF2AK2/MX2/HLA-E/IFI44/NMI/LGALS9/XRCC5/XBP1/IFI35/SP100/ZC3HAV1/CD55/GNLY/TAP1/HLA-A/HLA-B/ARID5A/PQBP1/TNFRSF4/PTPRC/NFKBIZ/FCER1G/STAT3/BST2/MPEG1/BTN2A2/CD38/JUNB/IFI16/CCL4/HLA-C/CD164/IL4I1/LILRA4/MYD88/ISG20/SLAMF7/IRF8/ADAR/APPL1/B2M/NUB1/KHDRBS1/ZNFX1/PML/CNPY3/DHX9/GPR65/TAPBP")

genes_24h <- format_gene_list("DUSP3/KIF5B/ARL8B/GNLY/MBP/NPLOC4/CXCL1/PPP2R3C/PHB2/CXCL3/EP300/DHX15/PYCARD/HLA-DMB/SPN/CD40/CD58/HLA-DRB1/MCOLN2/TFRC/PRDX1/AHR/PTPN2/BIRC3/PARP9/TUBB4B/PTGER4/ACTR3/PJA2/CD86/TP53/CYLD/LAMP3/DBNL/CCL22/PTPRC/SASH3/IFI16/TANK/GPX1/IFNGR1/CTSH/HCK/ICOSLG/UBE2N/HSP90AA1/IL7R/LY96/ALCAM/HLA-DQA2/CDC42/IL4I1/CD83/CD55/BAX/TNFRSF4/IFNAR1/SETD2/LAPTM5/DDX3X/EBI3/GSN/HLA-DOA/CSF2RB/CST7/SAMHD1/CD70/LTB")

genes <- unique(c(genes_24h, genes_3h, genes_UT))

heatmap_immu__UT <- DoHeatmap(cluster.average_UT, features = genes, group.by = 'sex', draw.lines = TRUE) +
 scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) + guides(fill = FALSE, color = FALSE) + ggtitle("UT") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y =element_text(size =10, face = "bold", vjust = 0),plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_immu__3hCA <- DoHeatmap(cluster.average_3hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(fill = FALSE, color = FALSE) + ggtitle("3hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

heatmap_immu__24hCA <- DoHeatmap(cluster.average_24hCA, features = genes, group.by = 'sex', draw.lines = TRUE) +
  scale_fill_gradientn(colors = brewer.pal(10, "RdBu"), oob = scales::oob_squish_any, limits = c(-2, 2)) +
  guides(color = FALSE) + ggtitle("24hCA") +
  theme(plot.title = element_text(hjust = 0.5), axis.text.y = element_blank(), plot.margin = unit(c(1, 0, 1, 1), "cm"))

grid.arrange(heatmap_immu__UT, heatmap_immu__3hCA, heatmap_immu__24hCA, ncol = 3)

```